INFO:root:PyLint Starting | Path: assignment/ | Threshold: 0.0 ,,,,
************* Module adhoc.cleanup_checkpoint_dirs,,,,
"assignment/adhoc/cleanup_checkpoint_dirs.py:11:0: C0413: Import ""import marty.common.python.io.s3 as io_s3"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/adhoc/cleanup_checkpoint_dirs.py:12:0: C0413: Import ""from marty.lib.job_manager import JobManager"" should be placed at the top of the module (wrong-import-position)",,,,
assignment/adhoc/cleanup_checkpoint_dirs.py:15:8: W0621: Redefining name 'job' from outer scope (line 91) (redefined-outer-name),,,,
************* Module adhoc.delete_campaign_data_from_dynamo,,,,
"assignment/adhoc/delete_campaign_data_from_dynamo.py:13:0: C0413: Import ""from marty.common.python.io.dynamo import dynamo_read_table"," dynamo_remove_from_table"" should be placed at the top of the module (wrong-import-position)",,,
"assignment/adhoc/delete_campaign_data_from_dynamo.py:17:0: C0413: Import ""from marty.lib.job_manager import JobManager"" should be placed at the top of the module (wrong-import-position)",,,,
assignment/adhoc/delete_campaign_data_from_dynamo.py:20:8: W0621: Redefining name 'job' from outer scope (line 80) (redefined-outer-name),,,,
************* Module adhoc.forced_cell_assignment_rerun,,,,
assignment/adhoc/forced_cell_assignment_rerun.py:12:0: E0401: Unable to import 'pyspark.sql.functions' (import-error),,,,
"assignment/adhoc/forced_cell_assignment_rerun.py:12:0: C0413: Import ""import pyspark.sql.functions as sqlf"" should be placed at the top of the module (wrong-import-position)",,,,
assignment/adhoc/forced_cell_assignment_rerun.py:13:0: E0401: Unable to import 'pyspark.sql.types' (import-error),,,,
"assignment/adhoc/forced_cell_assignment_rerun.py:13:0: C0413: Import ""import pyspark.sql.types as sqlt"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/adhoc/forced_cell_assignment_rerun.py:15:0: C0413: Import ""from marty.assignment.lib.cell_assignment import forced_cell_assignment"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/adhoc/forced_cell_assignment_rerun.py:16:0: C0413: Import ""from marty.common.python.io.dynamo import dynamo_read_table"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/adhoc/forced_cell_assignment_rerun.py:17:0: C0413: Import ""from marty.lib.job_manager import JobManager"" should be placed at the top of the module (wrong-import-position)",,,,
assignment/adhoc/forced_cell_assignment_rerun.py:21:4: W0621: Redefining name 'job' from outer scope (line 213) (redefined-outer-name),,,,
assignment/adhoc/forced_cell_assignment_rerun.py:21:9: W0621: Redefining name 'forced_cell_assignment' from outer scope (line 15) (redefined-outer-name),,,,
assignment/adhoc/forced_cell_assignment_rerun.py:20:0: W0102: Dangerous default value [] as argument (dangerous-default-value),,,,
assignment/adhoc/forced_cell_assignment_rerun.py:42:8: W0621: Redefining name 'job' from outer scope (line 213) (redefined-outer-name),,,,
assignment/adhoc/forced_cell_assignment_rerun.py:42:0: W0102: Dangerous default value [] as argument (dangerous-default-value),,,,
assignment/adhoc/forced_cell_assignment_rerun.py:42:0: R0914: Too many local variables (18/15) (too-many-locals),,,,
"assignment/adhoc/forced_cell_assignment_rerun.py:12:0: C0411: third party import ""import pyspark.sql.functions as sqlf"" should be placed before ""from marty.common.python.io.logger import MartyLogger"" (wrong-import-order)",,,,
"assignment/adhoc/forced_cell_assignment_rerun.py:13:0: C0411: third party import ""import pyspark.sql.types as sqlt"" should be placed before ""from marty.common.python.io.logger import MartyLogger"" (wrong-import-order)",,,,
assignment/adhoc/forced_cell_assignment_rerun.py:15:0: C0412: Imports from package marty are not grouped (ungrouped-imports),,,,
************* Module adhoc.load_campaign_data_into_dynamo,,,,
"assignment/adhoc/load_campaign_data_into_dynamo.py:11:0: C0413: Import ""import marty.common.python.io.s3 as io_s3"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/adhoc/load_campaign_data_into_dynamo.py:12:0: C0413: Import ""from marty.common.python.io.dynamo import dynamo_add_to_table"," dynamo_read_table"" should be placed at the top of the module (wrong-import-position)",,,
"assignment/adhoc/load_campaign_data_into_dynamo.py:16:0: C0413: Import ""from marty.lib.job_manager import JobManager"" should be placed at the top of the module (wrong-import-position)",,,,
assignment/adhoc/load_campaign_data_into_dynamo.py:28:21: R1718: Consider using a set comprehension (consider-using-set-comprehension),,,,
assignment/adhoc/load_campaign_data_into_dynamo.py:34:26: R1718: Consider using a set comprehension (consider-using-set-comprehension),,,,
assignment/adhoc/load_campaign_data_into_dynamo.py:81:8: W0621: Redefining name 'job' from outer scope (line 140) (redefined-outer-name),,,,
assignment/adhoc/load_campaign_data_into_dynamo.py:11:0: W0611: Unused marty.common.python.io.s3 imported as io_s3 (unused-import),,,,
************* Module adhoc.past_paid_response_model_decile_migration_report,,,,
"assignment/adhoc/past_paid_response_model_decile_migration_report.py:14:0: C0413: Import ""import marty.assignment.lib.response_model_decile_migration_utils as report_utils"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/adhoc/past_paid_response_model_decile_migration_report.py:15:0: C0413: Import ""from marty.common.python.io.s3 import input_data_validator"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/adhoc/past_paid_response_model_decile_migration_report.py:16:0: C0413: Import ""from marty.lib.job_manager import JobManager"" should be placed at the top of the module (wrong-import-position)",,,,
assignment/adhoc/past_paid_response_model_decile_migration_report.py:19:8: W0621: Redefining name 'job' from outer scope (line 91) (redefined-outer-name),,,,
************* Module adhoc.prospect_response_model_decile_migration_report,,,,
"assignment/adhoc/prospect_response_model_decile_migration_report.py:14:0: C0413: Import ""import marty.assignment.lib.response_model_decile_migration_utils as report_utils"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/adhoc/prospect_response_model_decile_migration_report.py:15:0: C0413: Import ""from marty.common.python.io.s3 import input_data_validator"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/adhoc/prospect_response_model_decile_migration_report.py:16:0: C0413: Import ""from marty.lib.job_manager import JobManager"" should be placed at the top of the module (wrong-import-position)",,,,
assignment/adhoc/prospect_response_model_decile_migration_report.py:19:8: W0621: Redefining name 'job' from outer scope (line 90) (redefined-outer-name),,,,
************* Module adhoc.qc_report,,,,
"assignment/adhoc/qc_report.py:11:0: C0413: Import ""import marty.assignment.lib.qc_generate_report as qc_report"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/adhoc/qc_report.py:12:0: C0413: Import ""from marty.common.python.io.s3 import input_data_validator"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/adhoc/qc_report.py:13:0: C0413: Import ""from marty.lib.job_manager import JobManager"" should be placed at the top of the module (wrong-import-position)",,,,
assignment/adhoc/qc_report.py:16:8: W0621: Redefining name 'job' from outer scope (line 109) (redefined-outer-name),,,,
************* Module adhoc.qc_report_final,,,,
assignment/adhoc/qc_report_final.py:11:0: E0401: Unable to import 'pyspark.sql.functions' (import-error),,,,
"assignment/adhoc/qc_report_final.py:11:0: C0413: Import ""import pyspark.sql.functions as sqlf"" should be placed at the top of the module (wrong-import-position)",,,,
assignment/adhoc/qc_report_final.py:12:0: E0401: Unable to import 'pyspark.sql.types' (import-error),,,,
"assignment/adhoc/qc_report_final.py:12:0: C0413: Import ""import pyspark.sql.types as sqlt"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/adhoc/qc_report_final.py:14:0: C0413: Import ""import marty.assignment.lib.qc_generate_final_report as generate_final_report"" should be placed at the top of the module (wrong-import-position)",,,,
assignment/adhoc/qc_report_final.py:15:0: R0402: Use 'from marty.common.python.io import excel_report' instead (consider-using-from-import),,,,
"assignment/adhoc/qc_report_final.py:15:0: C0413: Import ""import marty.common.python.io.excel_report as excel_report"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/adhoc/qc_report_final.py:16:0: C0413: Import ""import marty.common.python.io.s3 as io_s3"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/adhoc/qc_report_final.py:17:0: C0413: Import ""from marty.common.python.io.s3 import input_data_validator"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/adhoc/qc_report_final.py:18:0: C0413: Import ""from marty.lib.job_manager import JobManager"" should be placed at the top of the module (wrong-import-position)",,,,
assignment/adhoc/qc_report_final.py:128:8: W0621: Redefining name 'job' from outer scope (line 404) (redefined-outer-name),,,,
assignment/adhoc/qc_report_final.py:128:0: R0914: Too many local variables (28/15) (too-many-locals),,,,
assignment/adhoc/qc_report_final.py:15:0: W0611: Unused marty.common.python.io.excel_report imported as excel_report (unused-import),,,,
assignment/adhoc/qc_report_final.py:16:0: W0611: Unused marty.common.python.io.s3 imported as io_s3 (unused-import),,,,
"assignment/adhoc/qc_report_final.py:11:0: C0411: third party import ""import pyspark.sql.functions as sqlf"" should be placed before ""from marty.common.python.io.logger import MartyLogger"" (wrong-import-order)",,,,
"assignment/adhoc/qc_report_final.py:12:0: C0411: third party import ""import pyspark.sql.types as sqlt"" should be placed before ""from marty.common.python.io.logger import MartyLogger"" (wrong-import-order)",,,,
************* Module adhoc.qc_report_second_suppression,,,,
assignment/adhoc/qc_report_second_suppression.py:102:0: C0301: Line too long (108/100) (line-too-long),,,,
assignment/adhoc/qc_report_second_suppression.py:117:0: C0301: Line too long (103/100) (line-too-long),,,,
"assignment/adhoc/qc_report_second_suppression.py:11:0: C0413: Import ""import boto3"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/adhoc/qc_report_second_suppression.py:12:0: C0413: Import ""import pandas as pd"" should be placed at the top of the module (wrong-import-position)",,,,
assignment/adhoc/qc_report_second_suppression.py:13:0: E0401: Unable to import 'pyspark.sql.functions' (import-error),,,,
"assignment/adhoc/qc_report_second_suppression.py:13:0: C0413: Import ""import pyspark.sql.functions as sqlf"" should be placed at the top of the module (wrong-import-position)",,,,
assignment/adhoc/qc_report_second_suppression.py:14:0: E0401: Unable to import 'pyspark.sql' (import-error),,,,
"assignment/adhoc/qc_report_second_suppression.py:14:0: C0413: Import ""from pyspark.sql import Window as sqlw"" should be placed at the top of the module (wrong-import-position)",,,,
assignment/adhoc/qc_report_second_suppression.py:15:0: E0401: Unable to import 'pyspark.storagelevel' (import-error),,,,
"assignment/adhoc/qc_report_second_suppression.py:15:0: C0413: Import ""from pyspark.storagelevel import StorageLevel"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/adhoc/qc_report_second_suppression.py:17:0: C0413: Import ""import marty.assignment.lib.utils as assignment_utils"" should be placed at the top of the module (wrong-import-position)",,,,
assignment/adhoc/qc_report_second_suppression.py:18:0: R0402: Use 'from marty.common.python.io import excel_report' instead (consider-using-from-import),,,,
"assignment/adhoc/qc_report_second_suppression.py:18:0: C0413: Import ""import marty.common.python.io.excel_report as excel_report"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/adhoc/qc_report_second_suppression.py:19:0: C0413: Import ""import marty.common.python.io.s3 as io_s3"" should be placed at the top of the module (wrong-import-position)",,,,
assignment/adhoc/qc_report_second_suppression.py:20:0: R0402: Use 'from marty.common.python.spark.misc import utils' instead (consider-using-from-import),,,,
"assignment/adhoc/qc_report_second_suppression.py:20:0: C0413: Import ""import marty.common.python.spark.misc.utils as utils"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/adhoc/qc_report_second_suppression.py:21:0: C0413: Import ""from marty.assignment.lib.pre_selection_utils import preprocess_suppression_daily"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/adhoc/qc_report_second_suppression.py:24:0: C0413: Import ""from marty.assignment.lib.second_suppression_and_segmentation_utils import _suppression_step", add_drive_time, add_ebt_eligible_flag, apply_nugget_exclusions," define_segments"" should be placed at the top of the module (wrong-import-position)"
"assignment/adhoc/qc_report_second_suppression.py:31:0: C0413: Import ""from marty.assignment.lib.uplift_utils import apply_uplift_exclusions"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/adhoc/qc_report_second_suppression.py:32:0: C0413: Import ""from marty.common.python.io.s3 import input_data_validator"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/adhoc/qc_report_second_suppression.py:33:0: C0413: Import ""from marty.lib.job_manager import JobManager"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/adhoc/qc_report_second_suppression.py:53:22: C0103: Argument name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/adhoc/qc_report_second_suppression.py:149:8: W0621: Redefining name 'job' from outer scope (line 486) (redefined-outer-name),,,,
assignment/adhoc/qc_report_second_suppression.py:149:0: R0914: Too many local variables (30/15) (too-many-locals),,,,
assignment/adhoc/qc_report_second_suppression.py:149:0: R0915: Too many statements (56/50) (too-many-statements),,,,
assignment/adhoc/qc_report_second_suppression.py:18:0: W0611: Unused marty.common.python.io.excel_report imported as excel_report (unused-import),,,,
"assignment/adhoc/qc_report_second_suppression.py:11:0: C0411: third party import ""import boto3"" should be placed before ""from marty.common.python.io.logger import MartyLogger"" (wrong-import-order)",,,,
"assignment/adhoc/qc_report_second_suppression.py:12:0: C0411: third party import ""import pandas as pd"" should be placed before ""from marty.common.python.io.logger import MartyLogger"" (wrong-import-order)",,,,
"assignment/adhoc/qc_report_second_suppression.py:13:0: C0411: third party import ""import pyspark.sql.functions as sqlf"" should be placed before ""from marty.common.python.io.logger import MartyLogger"" (wrong-import-order)",,,,
"assignment/adhoc/qc_report_second_suppression.py:14:0: C0411: third party import ""from pyspark.sql import Window as sqlw"" should be placed before ""from marty.common.python.io.logger import MartyLogger"" (wrong-import-order)",,,,
"assignment/adhoc/qc_report_second_suppression.py:15:0: C0411: third party import ""from pyspark.storagelevel import StorageLevel"" should be placed before ""from marty.common.python.io.logger import MartyLogger"" (wrong-import-order)",,,,
************* Module adhoc.response_ltv_optimization_comparison_and_profiling_final,,,,
assignment/adhoc/response_ltv_optimization_comparison_and_profiling_final.py:9:0: E0401: Unable to import 'pyspark.sql.functions' (import-error),,,,
assignment/adhoc/response_ltv_optimization_comparison_and_profiling_final.py:10:0: E0401: Unable to import 'pyspark.sql.types' (import-error),,,,
assignment/adhoc/response_ltv_optimization_comparison_and_profiling_final.py:17:0: R0402: Use 'from marty.common.python.io import excel_report' instead (consider-using-from-import),,,,
"assignment/adhoc/response_ltv_optimization_comparison_and_profiling_final.py:17:0: C0413: Import ""import marty.common.python.io.excel_report as excel_report"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/adhoc/response_ltv_optimization_comparison_and_profiling_final.py:18:0: C0413: Import ""import marty.common.python.io.s3 as io_s3"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/adhoc/response_ltv_optimization_comparison_and_profiling_final.py:19:0: C0413: Import ""from marty.common.python.io.s3 import input_data_validator"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/adhoc/response_ltv_optimization_comparison_and_profiling_final.py:20:0: C0413: Import ""from marty.lib.job_manager import JobManager"" should be placed at the top of the module (wrong-import-position)",,,,
assignment/adhoc/response_ltv_optimization_comparison_and_profiling_final.py:106:8: W0621: Redefining name 'job' from outer scope (line 273) (redefined-outer-name),,,,
assignment/adhoc/response_ltv_optimization_comparison_and_profiling_final.py:114:4: W0612: Unused variable 'campaign_name' (unused-variable),,,,
assignment/adhoc/response_ltv_optimization_comparison_and_profiling_final.py:17:0: W0611: Unused marty.common.python.io.excel_report imported as excel_report (unused-import),,,,
assignment/adhoc/response_ltv_optimization_comparison_and_profiling_final.py:18:0: W0611: Unused marty.common.python.io.s3 imported as io_s3 (unused-import),,,,
************* Module coverage.test_coverage,,,,
assignment/coverage/test_coverage.py:5:0: C0301: Line too long (111/100) (line-too-long),,,,
assignment/coverage/test_coverage.py:13:0: C0301: Line too long (101/100) (line-too-long),,,,
assignment/coverage/test_coverage.py:15:0: C0301: Line too long (119/100) (line-too-long),,,,
assignment/coverage/test_coverage.py:5:0: R0402: Use 'from marty.assignment.tests.unit import campaign_master_table_validator_unit' instead (consider-using-from-import),,,,
assignment/coverage/test_coverage.py:6:0: R0402: Use 'from marty.assignment.tests.unit import check_input_unit' instead (consider-using-from-import),,,,
assignment/coverage/test_coverage.py:7:0: R0402: Use 'from marty.assignment.tests.unit import final_dq_checks_unit' instead (consider-using-from-import),,,,
assignment/coverage/test_coverage.py:8:0: R0402: Use 'from marty.assignment.tests.unit import forced_cell_assignment_unit' instead (consider-using-from-import),,,,
assignment/coverage/test_coverage.py:9:0: R0402: Use 'from marty.assignment.tests.unit import main_cell_eligible_unit' instead (consider-using-from-import),,,,
assignment/coverage/test_coverage.py:10:0: R0402: Use 'from marty.assignment.tests.unit import main_cell_randomization_unit' instead (consider-using-from-import),,,,
assignment/coverage/test_coverage.py:11:0: R0402: Use 'from marty.assignment.tests.unit import offer_utils_unit' instead (consider-using-from-import),,,,
assignment/coverage/test_coverage.py:12:0: R0402: Use 'from marty.assignment.tests.unit import pcoa_utils_unit' instead (consider-using-from-import),,,,
assignment/coverage/test_coverage.py:13:0: R0402: Use 'from marty.assignment.tests.unit import pre_selection_intermediate_unit' instead (consider-using-from-import),,,,
assignment/coverage/test_coverage.py:14:0: R0402: Use 'from marty.assignment.tests.unit import pre_selection_unit' instead (consider-using-from-import),,,,
assignment/coverage/test_coverage.py:15:0: R0402: Use 'from marty.assignment.tests.unit import second_suppression_and_segmentation_unit' instead (consider-using-from-import),,,,
assignment/coverage/test_coverage.py:16:0: R0402: Use 'from marty.assignment.tests.unit import segments_unit' instead (consider-using-from-import),,,,
assignment/coverage/test_coverage.py:17:0: R0402: Use 'from marty.assignment.tests.unit import uplift_utils_unit' instead (consider-using-from-import),,,,
assignment/coverage/test_coverage.py:18:0: R0402: Use 'from marty.assignment.tests.unit import utuc_unit' instead (consider-using-from-import),,,,
************* Module lib.assign_utils,,,,
assignment/lib/assign_utils.py:94:32: W1401: Anomalous backslash in string: '\s'. String constant might be missing an r prefix. (anomalous-backslash-in-string),,,,
assignment/lib/assign_utils.py:94:39: W1401: Anomalous backslash in string: '\s'. String constant might be missing an r prefix. (anomalous-backslash-in-string),,,,
assignment/lib/assign_utils.py:94:45: W1401: Anomalous backslash in string: '\s'. String constant might be missing an r prefix. (anomalous-backslash-in-string),,,,
assignment/lib/assign_utils.py:94:48: W1401: Anomalous backslash in string: '\('. String constant might be missing an r prefix. (anomalous-backslash-in-string),,,,
assignment/lib/assign_utils.py:94:50: W1401: Anomalous backslash in string: '\{'. String constant might be missing an r prefix. (anomalous-backslash-in-string),,,,
assignment/lib/assign_utils.py:96:19: W1401: Anomalous backslash in string: '\}'. String constant might be missing an r prefix. (anomalous-backslash-in-string),,,,
assignment/lib/assign_utils.py:96:21: W1401: Anomalous backslash in string: '\)'. String constant might be missing an r prefix. (anomalous-backslash-in-string),,,,
assignment/lib/assign_utils.py:105:32: W1401: Anomalous backslash in string: '\s'. String constant might be missing an r prefix. (anomalous-backslash-in-string),,,,
assignment/lib/assign_utils.py:105:40: W1401: Anomalous backslash in string: '\s'. String constant might be missing an r prefix. (anomalous-backslash-in-string),,,,
assignment/lib/assign_utils.py:105:52: W1401: Anomalous backslash in string: '\s'. String constant might be missing an r prefix. (anomalous-backslash-in-string),,,,
assignment/lib/assign_utils.py:105:55: W1401: Anomalous backslash in string: '\('. String constant might be missing an r prefix. (anomalous-backslash-in-string),,,,
assignment/lib/assign_utils.py:105:57: W1401: Anomalous backslash in string: '\{'. String constant might be missing an r prefix. (anomalous-backslash-in-string),,,,
assignment/lib/assign_utils.py:107:19: W1401: Anomalous backslash in string: '\}'. String constant might be missing an r prefix. (anomalous-backslash-in-string),,,,
assignment/lib/assign_utils.py:107:21: W1401: Anomalous backslash in string: '\)'. String constant might be missing an r prefix. (anomalous-backslash-in-string),,,,
assignment/lib/assign_utils.py:7:0: E0401: Unable to import 'pyspark.sql.functions' (import-error),,,,
assignment/lib/assign_utils.py:8:0: E0401: Unable to import 'pyspark.sql.types' (import-error),,,,
assignment/lib/assign_utils.py:35:0: R0914: Too many local variables (18/15) (too-many-locals),,,,
assignment/lib/assign_utils.py:65:16: W1309: Using an f-string that does not have any interpolated variables (f-string-without-interpolation),,,,
assignment/lib/assign_utils.py:66:18: W1309: Using an f-string that does not have any interpolated variables (f-string-without-interpolation),,,,
"assignment/lib/assign_utils.py:75:11: R1714: Consider merging these comparisons with ""in"" to ""parametrized_key in ('bjs_cell'"," 'merkle_cell')"" (consider-using-in)",,,
assignment/lib/assign_utils.py:92:12: R1721: Unnecessary use of a comprehension, use list(re.finditer('([a-zA-Z0-9_]+)\\s+(not\\s+in)\\s+\\(\\{' + parametrized_key + '\\}\\)', segment_filter_condition)) instead. (unnecessary-comprehension),,
assignment/lib/assign_utils.py:103:12: R1721: Unnecessary use of a comprehension, use list(re.finditer('([a-zA-Z0-9_]+)\\s+((not\\s+in)|(in))\\s+\\(\\{' + parametrized_key + '\\}\\)', segment_filter_condition)) instead. (unnecessary-comprehension),,
assignment/lib/assign_utils.py:113:61: W0640: Cell variable not_in_clause defined in loop (cell-var-from-loop),,,,
assignment/lib/assign_utils.py:121:18: W1309: Using an f-string that does not have any interpolated variables (f-string-without-interpolation),,,,
assignment/lib/assign_utils.py:35:0: R0912: Too many branches (16/12) (too-many-branches),,,,
"assignment/lib/assign_utils.py:254:8: C0103: Variable name ""r"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/lib/assign_utils.py:7:0: W0611: Unused pyspark.sql.functions imported as sqlf (unused-import),,,,
assignment/lib/assign_utils.py:8:0: W0611: Unused pyspark.sql.types imported as sqlt (unused-import),,,,
************* Module lib.assignment_utuc,,,,
assignment/lib/assignment_utuc.py:9:0: E0401: Unable to import 'pyspark.sql.functions' (import-error),,,,
assignment/lib/assignment_utuc.py:10:0: E0401: Unable to import 'pyspark.sql.types' (import-error),,,,
assignment/lib/assignment_utuc.py:11:0: E0401: Unable to import 'pyspark.sql' (import-error),,,,
assignment/lib/assignment_utuc.py:71:0: R0914: Too many local variables (17/15) (too-many-locals),,,,
assignment/lib/assignment_utuc.py:99:16: W1309: Using an f-string that does not have any interpolated variables (f-string-without-interpolation),,,,
"assignment/lib/assignment_utuc.py:9:0: C0411: third party import ""import pyspark.sql.functions as sqlf"" should be placed before ""import marty.assignment.lib.utils as assignment_utils"" (wrong-import-order)",,,,
"assignment/lib/assignment_utuc.py:10:0: C0411: third party import ""import pyspark.sql.types as sqlt"" should be placed before ""import marty.assignment.lib.utils as assignment_utils"" (wrong-import-order)",,,,
"assignment/lib/assignment_utuc.py:11:0: C0411: third party import ""from pyspark.sql import Window as sqlw"" should be placed before ""import marty.assignment.lib.utils as assignment_utils"" (wrong-import-order)",,,,
************* Module lib.campaign_master_table_validator,,,,
"assignment/lib/campaign_master_table_validator.py:18:4: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/lib/campaign_master_table_validator.py:17:4: W0612: Unused variable 'unique_columns_set' (unused-variable),,,,
assignment/lib/campaign_master_table_validator.py:79:20: W0707: Consider explicitly re-raising using the 'from' keyword (raise-missing-from),,,,
************* Module lib.cell_assignment,,,,
assignment/lib/cell_assignment.py:805:29: W0511: TODO: add roi (fixme),,,,
assignment/lib/cell_assignment.py:7:0: E0401: Unable to import 'pyspark.sql.functions' (import-error),,,,
assignment/lib/cell_assignment.py:8:0: E0401: Unable to import 'pyspark.sql.types' (import-error),,,,
assignment/lib/cell_assignment.py:9:0: R0402: Use 'from scipy import stats' instead (consider-using-from-import),,,,
assignment/lib/cell_assignment.py:10:0: E0401: Unable to import 'pyspark.sql' (import-error),,,,
"assignment/lib/cell_assignment.py:22:24: C0103: Argument name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/lib/cell_assignment.py:42:27: C0103: Argument name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/lib/cell_assignment.py:89:0: R0913: Too many arguments (10/5) (too-many-arguments),,,,
assignment/lib/cell_assignment.py:275:0: R0913: Too many arguments (7/5) (too-many-arguments),,,,
assignment/lib/cell_assignment.py:314:0: R0913: Too many arguments (8/5) (too-many-arguments),,,,
assignment/lib/cell_assignment.py:389:0: R0913: Too many arguments (7/5) (too-many-arguments),,,,
assignment/lib/cell_assignment.py:389:0: R0914: Too many local variables (40/15) (too-many-locals),,,,
assignment/lib/cell_assignment.py:488:16: W1203: Use lazy % formatting in logging functions (logging-fstring-interpolation),,,,
assignment/lib/cell_assignment.py:389:0: R0912: Too many branches (30/12) (too-many-branches),,,,
assignment/lib/cell_assignment.py:389:0: R0915: Too many statements (90/50) (too-many-statements),,,,
assignment/lib/cell_assignment.py:765:0: R0913: Too many arguments (6/5) (too-many-arguments),,,,
assignment/lib/cell_assignment.py:765:0: R0914: Too many local variables (18/15) (too-many-locals),,,,
assignment/lib/cell_assignment.py:843:16: W1203: Use lazy % formatting in logging functions (logging-fstring-interpolation),,,,
assignment/lib/cell_assignment.py:850:16: E1205: Too many arguments for logging format string (logging-too-many-args),,,,
************* Module lib.file_transfer_utils,,,,
"assignment/lib/file_transfer_utils.py:57:4: R1705: Unnecessary ""else"" after ""return"""," remove the ""else"" and de-indent the code inside it (no-else-return)",,,
"assignment/lib/file_transfer_utils.py:72:8: C0103: Variable name ""r"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/lib/file_transfer_utils.py:4:0: W0611: Unused datetime imported as dt (unused-import),,,,
************* Module lib.final_dq_checks,,,,
assignment/lib/final_dq_checks.py:5:0: E0401: Unable to import 'pyspark.sql.functions' (import-error),,,,
assignment/lib/final_dq_checks.py:6:0: E0401: Unable to import 'pyspark.sql.types' (import-error),,,,
assignment/lib/final_dq_checks.py:7:0: E0401: Unable to import 'pyspark.sql' (import-error),,,,
assignment/lib/final_dq_checks.py:8:0: E0401: Unable to import 'pyspark.sql.functions' (import-error),,,,
assignment/lib/final_dq_checks.py:45:4: W0102: Dangerous default value {} as argument (dangerous-default-value),,,,
assignment/lib/final_dq_checks.py:45:4: R0913: Too many arguments (6/5) (too-many-arguments),,,,
assignment/lib/final_dq_checks.py:94:4: W0102: Dangerous default value {} as argument (dangerous-default-value),,,,
assignment/lib/final_dq_checks.py:94:4: R0913: Too many arguments (6/5) (too-many-arguments),,,,
"assignment/lib/final_dq_checks.py:135:8: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/lib/final_dq_checks.py:164:12: C0103: Variable name ""d"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/lib/final_dq_checks.py:214:4: W0102: Dangerous default value {} as argument (dangerous-default-value),,,,
assignment/lib/final_dq_checks.py:214:4: R0913: Too many arguments (6/5) (too-many-arguments),,,,
assignment/lib/final_dq_checks.py:244:4: R0914: Too many local variables (16/15) (too-many-locals),,,,
"assignment/lib/final_dq_checks.py:257:8: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/lib/final_dq_checks.py:269:12: C0103: Variable name ""c"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/lib/final_dq_checks.py:286:20: C0103: Variable name ""w"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/lib/final_dq_checks.py:350:8: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/lib/final_dq_checks.py:362:12: C0103: Variable name ""m"" doesn't conform to snake_case naming style (invalid-name)",,,,
************* Module lib.gurobi_input_preparation,,,,
assignment/lib/gurobi_input_preparation.py:5:0: E0401: Unable to import 'pyspark.sql' (import-error),,,,
assignment/lib/gurobi_input_preparation.py:6:0: E0401: Unable to import 'pyspark.sql.window' (import-error),,,,
"assignment/lib/gurobi_input_preparation.py:106:4: C0103: Variable name ""f1"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/lib/gurobi_input_preparation.py:107:4: C0103: Variable name ""f2"" doesn't conform to snake_case naming style (invalid-name)",,,,
************* Module lib.main_cell_eligible,,,,
assignment/lib/main_cell_eligible.py:7:0: E0401: Unable to import 'pyspark.sql.functions' (import-error),,,,
assignment/lib/main_cell_eligible.py:8:0: E0401: Unable to import 'pyspark.sql' (import-error),,,,
assignment/lib/main_cell_eligible.py:12:4: W0613: Unused argument 'job' (unused-argument),,,,
************* Module lib.model_scores,,,,
assignment/lib/model_scores.py:8:0: E0401: Unable to import 'pyspark.sql.functions' (import-error),,,,
assignment/lib/model_scores.py:9:0: E0401: Unable to import 'pyspark.sql.types' (import-error),,,,
assignment/lib/model_scores.py:256:12: W1309: Using an f-string that does not have any interpolated variables (f-string-without-interpolation),,,,
assignment/lib/model_scores.py:260:33: W1309: Using an f-string that does not have any interpolated variables (f-string-without-interpolation),,,,
assignment/lib/model_scores.py:263:12: W1309: Using an f-string that does not have any interpolated variables (f-string-without-interpolation),,,,
assignment/lib/model_scores.py:267:33: W1309: Using an f-string that does not have any interpolated variables (f-string-without-interpolation),,,,
************* Module lib.offer_utils,,,,
assignment/lib/offer_utils.py:7:0: E0401: Unable to import 'pyspark.sql.functions' (import-error),,,,
assignment/lib/offer_utils.py:8:0: E0401: Unable to import 'pyspark.sql.types' (import-error),,,,
assignment/lib/offer_utils.py:9:0: E0401: Unable to import 'pyspark.sql' (import-error),,,,
assignment/lib/offer_utils.py:14:0: R0914: Too many local variables (17/15) (too-many-locals),,,,
"assignment/lib/offer_utils.py:95:8: C0103: Variable name ""o"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/lib/offer_utils.py:110:8: C0103: Variable name ""o"" doesn't conform to snake_case naming style (invalid-name)",,,,
************* Module lib.pcoa_utils,,,,
assignment/lib/pcoa_utils.py:4:0: E0401: Unable to import 'pyspark.sql.functions' (import-error),,,,
assignment/lib/pcoa_utils.py:156:0: R0914: Too many local variables (20/15) (too-many-locals),,,,
************* Module lib.pre_selection_utils,,,,
assignment/lib/pre_selection_utils.py:330:0: C0301: Line too long (101/100) (line-too-long),,,,
assignment/lib/pre_selection_utils.py:386:0: C0301: Line too long (122/100) (line-too-long),,,,
assignment/lib/pre_selection_utils.py:8:0: E0401: Unable to import 'pyspark.sql.functions' (import-error),,,,
assignment/lib/pre_selection_utils.py:9:0: E0401: Unable to import 'pyspark.sql.types' (import-error),,,,
assignment/lib/pre_selection_utils.py:10:0: E0401: Unable to import 'pyspark.sql' (import-error),,,,
assignment/lib/pre_selection_utils.py:12:0: R0402: Use 'from marty.assignment.lib import offer_utils' instead (consider-using-from-import),,,,
assignment/lib/pre_selection_utils.py:106:0: R0914: Too many local variables (24/15) (too-many-locals),,,,
"assignment/lib/pre_selection_utils.py:146:8: C0103: Variable name ""c"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/lib/pre_selection_utils.py:161:42: C0121: Comparison 'sqlf.col('force_exclusion') == False' should be 'sqlf.col('force_exclusion') is False' if checking for the singleton value False, or 'not sqlf.col('force_exclusion')' if testing for falsiness (singleton-comparison),,,
"assignment/lib/pre_selection_utils.py:200:12: C0103: Variable name ""c"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/lib/pre_selection_utils.py:201:12: C0103: Variable name ""w"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/lib/pre_selection_utils.py:242:15: C0123: Use isinstance() rather than type() for a typecheck. (unidiomatic-typecheck),,,,
assignment/lib/pre_selection_utils.py:248:15: C0123: Use isinstance() rather than type() for a typecheck. (unidiomatic-typecheck),,,,
assignment/lib/pre_selection_utils.py:258:25: C0121: Comparison 'sqlf.col('force_exclusion') == True' should be 'sqlf.col('force_exclusion') is True' if checking for the singleton value True, or 'bool(sqlf.col('force_exclusion'))' if testing for truthiness (singleton-comparison),,,
assignment/lib/pre_selection_utils.py:106:0: R0912: Too many branches (15/12) (too-many-branches),,,,
assignment/lib/pre_selection_utils.py:280:0: R0914: Too many local variables (23/15) (too-many-locals),,,,
"assignment/lib/pre_selection_utils.py:315:12: C0103: Variable name ""c"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/lib/pre_selection_utils.py:335:12: C0103: Variable name ""c"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/lib/pre_selection_utils.py:360:12: C0103: Variable name ""c"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/lib/pre_selection_utils.py:280:0: R0912: Too many branches (13/12) (too-many-branches),,,,
assignment/lib/pre_selection_utils.py:390:0: R0913: Too many arguments (11/5) (too-many-arguments),,,,
assignment/lib/pre_selection_utils.py:390:0: R0914: Too many local variables (16/15) (too-many-locals),,,,
************* Module lib.qc_generate_final_report,,,,
assignment/lib/qc_generate_final_report.py:11:0: R0402: Use 'from matplotlib import colors' instead (consider-using-from-import),,,,
assignment/lib/qc_generate_final_report.py:13:0: R0402: Use 'from matplotlib import ticker' instead (consider-using-from-import),,,,
assignment/lib/qc_generate_final_report.py:16:0: E0401: Unable to import 'pyspark.sql.functions' (import-error),,,,
assignment/lib/qc_generate_final_report.py:17:0: E0401: Unable to import 'pyspark.sql' (import-error),,,,
assignment/lib/qc_generate_final_report.py:19:0: R0402: Use 'from marty.common.python.io import excel_report' instead (consider-using-from-import),,,,
"assignment/lib/qc_generate_final_report.py:55:8: C0103: Variable name ""x"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/lib/qc_generate_final_report.py:55:11: C0103: Variable name ""y"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/lib/qc_generate_final_report.py:74:17: R1735: Consider using {} instead of dict() (use-dict-literal),,,,
assignment/lib/qc_generate_final_report.py:75:14: R1735: Consider using {} instead of dict() (use-dict-literal),,,,
assignment/lib/qc_generate_final_report.py:76:13: R1735: Consider using {} instead of dict() (use-dict-literal),,,,
assignment/lib/qc_generate_final_report.py:78:23: R1735: Consider using {} instead of dict() (use-dict-literal),,,,
"assignment/lib/qc_generate_final_report.py:80:8: C0103: Variable name ""c"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/lib/qc_generate_final_report.py:82:12: C0103: Variable name ""g"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/lib/qc_generate_final_report.py:84:12: C0103: Variable name ""g"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/lib/qc_generate_final_report.py:88:11: C0113: Consider changing ""not i in indices"" to ""i not in indices"" (unneeded-not)",,,,
"assignment/lib/qc_generate_final_report.py:103:8: C0103: Variable name ""c"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/lib/qc_generate_final_report.py:105:12: C0103: Variable name ""pc"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/lib/qc_generate_final_report.py:107:12: C0103: Variable name ""pc"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/lib/qc_generate_final_report.py:110:8: C0103: Variable name ""c"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/lib/qc_generate_final_report.py:111:22: C0209: Formatting a regular string which could be a f-string (consider-using-f-string),,,,
assignment/lib/qc_generate_final_report.py:59:0: R0912: Too many branches (13/12) (too-many-branches),,,,
"assignment/lib/qc_generate_final_report.py:329:4: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/lib/qc_generate_final_report.py:331:4: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/lib/qc_generate_final_report.py:377:4: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/lib/qc_generate_final_report.py:427:4: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/lib/qc_generate_final_report.py:548:4: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/lib/qc_generate_final_report.py:591:4: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/lib/qc_generate_final_report.py:603:43: W1309: Using an f-string that does not have any interpolated variables (f-string-without-interpolation),,,,
"assignment/lib/qc_generate_final_report.py:625:4: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/lib/qc_generate_final_report.py:652:4: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/lib/qc_generate_final_report.py:712:4: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/lib/qc_generate_final_report.py:716:4: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/lib/qc_generate_final_report.py:717:4: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/lib/qc_generate_final_report.py:748:4: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/lib/qc_generate_final_report.py:750:7: C0103: Variable name ""ax"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/lib/qc_generate_final_report.py:770:4: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/lib/qc_generate_final_report.py:790:13: W0108: Lambda may not be necessary (unnecessary-lambda),,,,
"assignment/lib/qc_generate_final_report.py:792:4: C0103: Variable name ""fd"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/lib/qc_generate_final_report.py:792:4: W0612: Unused variable 'fd' (unused-variable),,,,
"assignment/lib/qc_generate_final_report.py:871:4: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/lib/qc_generate_final_report.py:873:4: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/lib/qc_generate_final_report.py:891:4: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/lib/qc_generate_final_report.py:923:4: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/lib/qc_generate_final_report.py:943:4: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
************* Module lib.qc_generate_report,,,,
assignment/lib/qc_generate_report.py:684:0: C0301: Line too long (105/100) (line-too-long),,,,
assignment/lib/qc_generate_report.py:9:0: E0401: Unable to import 'pyspark.sql.functions' (import-error),,,,
"assignment/lib/qc_generate_report.py:289:24: C0103: Argument name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/lib/qc_generate_report.py:493:4: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
************* Module lib.response_model_decile_migration_utils,,,,
assignment/lib/response_model_decile_migration_utils.py:35:13: W1401: Anomalous backslash in string: '\/'. String constant might be missing an r prefix. (anomalous-backslash-in-string),,,,
assignment/lib/response_model_decile_migration_utils.py:35:30: W1401: Anomalous backslash in string: '\d'. String constant might be missing an r prefix. (anomalous-backslash-in-string),,,,
assignment/lib/response_model_decile_migration_utils.py:35:37: W1401: Anomalous backslash in string: '\d'. String constant might be missing an r prefix. (anomalous-backslash-in-string),,,,
assignment/lib/response_model_decile_migration_utils.py:35:44: W1401: Anomalous backslash in string: '\d'. String constant might be missing an r prefix. (anomalous-backslash-in-string),,,,
assignment/lib/response_model_decile_migration_utils.py:10:0: E0401: Unable to import 'pyspark.ml.feature' (import-error),,,,
assignment/lib/response_model_decile_migration_utils.py:11:0: E0401: Unable to import 'pyspark.sql' (import-error),,,,
assignment/lib/response_model_decile_migration_utils.py:42:12: R1718: Consider using a set comprehension (consider-using-set-comprehension),,,,
"assignment/lib/response_model_decile_migration_utils.py:57:4: C0103: Variable name ""d"" doesn't conform to snake_case naming style (invalid-name)",,,,
************* Module lib.second_suppression_and_segmentation_utils,,,,
assignment/lib/second_suppression_and_segmentation_utils.py:5:0: E0401: Unable to import 'pyspark.ml.feature' (import-error),,,,
assignment/lib/second_suppression_and_segmentation_utils.py:6:0: E0401: Unable to import 'pyspark.sql.functions' (import-error),,,,
assignment/lib/second_suppression_and_segmentation_utils.py:7:0: E0401: Unable to import 'pyspark.sql.types' (import-error),,,,
assignment/lib/second_suppression_and_segmentation_utils.py:8:0: E0401: Unable to import 'pyspark.sql' (import-error),,,,
assignment/lib/second_suppression_and_segmentation_utils.py:12:0: R0402: Use 'from marty.common.python.spark.misc import utils' instead (consider-using-from-import),,,,
assignment/lib/second_suppression_and_segmentation_utils.py:190:4: W1203: Use lazy % formatting in logging functions (logging-fstring-interpolation),,,,
"assignment/lib/second_suppression_and_segmentation_utils.py:468:20: C0103: Argument name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
************* Module lib.uplift_utils,,,,
assignment/lib/uplift_utils.py:4:0: E0401: Unable to import 'pyspark.sql.functions' (import-error),,,,
"assignment/lib/uplift_utils.py:12:28: C0103: Argument name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/lib/uplift_utils.py:35:4: W1203: Use lazy % formatting in logging functions (logging-fstring-interpolation),,,,
************* Module lib.utils,,,,
assignment/lib/utils.py:8:0: R0402: Use 'from dateutil import relativedelta' instead (consider-using-from-import),,,,
assignment/lib/utils.py:9:0: E0401: Unable to import 'pyspark.sql.functions' (import-error),,,,
"assignment/lib/utils.py:83:20: C0103: Argument name ""x"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/lib/utils.py:97:7: C0123: Use isinstance() rather than type() for a typecheck. (unidiomatic-typecheck),,,,
assignment/lib/utils.py:97:27: C0123: Use isinstance() rather than type() for a typecheck. (unidiomatic-typecheck),,,,
"assignment/lib/utils.py:205:4: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/lib/utils.py:230:4: W0102: Dangerous default value [] as argument (dangerous-default-value),,,,
"assignment/lib/utils.py:231:8: C0103: Argument name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/lib/utils.py:233:25: C0103: Argument name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/lib/utils.py:254:4: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/lib/utils.py:270:12: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/lib/utils.py:281:4: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/lib/utils.py:292:4: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/lib/utils.py:383:12: W1309: Using an f-string that does not have any interpolated variables (f-string-without-interpolation),,,,
************* Module lib.input_checks.config_checker,,,,
assignment/lib/input_checks/config_checker.py:69:9: W1401: Anomalous backslash in string: '\*'. String constant might be missing an r prefix. (anomalous-backslash-in-string),,,,
assignment/lib/input_checks/config_checker.py:70:9: W1401: Anomalous backslash in string: '\+'. String constant might be missing an r prefix. (anomalous-backslash-in-string),,,,
"assignment/lib/input_checks/config_checker.py:59:4: C0103: Variable name ""KEYWORDS"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/lib/input_checks/config_checker.py:74:18: C0103: Argument name ""s"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/lib/input_checks/config_checker.py:135:17: W0621: Redefining name 'job' from outer scope (line 164) (redefined-outer-name),,,,
************* Module lib.tests.unit.qc_final_report_test,,,,
assignment/lib/tests/unit/qc_final_report_test.py:7:0: E0401: Unable to import 'pyspark.sql.types' (import-error),,,,
assignment/lib/tests/unit/qc_final_report_test.py:20:4: W0221: Number of parameters was 1 in 'TestCase.setUp' and is now 1 in overridden 'TestQcFinalReportUtils.setUp' method (arguments-differ),,,,
************* Module lib.tests.unit.utils_test,,,,
assignment/lib/tests/unit/utils_test.py:9:0: E0401: Unable to import 'pyspark.sql.types' (import-error),,,,
assignment/lib/tests/unit/utils_test.py:13:0: R0402: Use 'from marty.lib import test_utils' instead (consider-using-from-import),,,,
assignment/lib/tests/unit/utils_test.py:23:4: W0221: Number of parameters was 1 in 'TestCase.setUp' and is now 1 in overridden 'TestAssignmentUtils.setUp' method (arguments-differ),,,,
"assignment/lib/tests/unit/utils_test.py:75:12: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/lib/tests/unit/utils_test.py:75:16: W0612: Unused variable 'counts' (unused-variable),,,,
"assignment/lib/tests/unit/utils_test.py:80:8: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/lib/tests/unit/utils_test.py:107:12: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/lib/tests/unit/utils_test.py:107:16: W0612: Unused variable 'counts' (unused-variable),,,,
"assignment/lib/tests/unit/utils_test.py:112:8: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/lib/tests/unit/utils_test.py:139:12: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/lib/tests/unit/utils_test.py:139:16: W0612: Unused variable 'counts' (unused-variable),,,,
"assignment/lib/tests/unit/utils_test.py:144:8: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/lib/tests/unit/utils_test.py:171:12: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/lib/tests/unit/utils_test.py:171:16: W0612: Unused variable 'counts' (unused-variable),,,,
"assignment/lib/tests/unit/utils_test.py:176:8: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/lib/tests/unit/utils_test.py:221:12: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/lib/tests/unit/utils_test.py:221:16: W0612: Unused variable 'counts' (unused-variable),,,,
"assignment/lib/tests/unit/utils_test.py:226:8: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/lib/tests/unit/utils_test.py:341:8: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
************* Module notebooks.dq_lib,,,,
assignment/notebooks/dq_lib.py:69:0: C0301: Line too long (166/100) (line-too-long),,,,
assignment/notebooks/dq_lib.py:157:0: C0301: Line too long (106/100) (line-too-long),,,,
assignment/notebooks/dq_lib.py:449:0: C0301: Line too long (121/100) (line-too-long),,,,
assignment/notebooks/dq_lib.py:1:0: C0114: Missing module docstring (missing-module-docstring),,,,
assignment/notebooks/dq_lib.py:10:0: E0401: Unable to import 'IPython.display' (import-error),,,,
"assignment/notebooks/dq_lib.py:10:0: C0413: Import ""from IPython.display import display"," HTML"" should be placed at the top of the module (wrong-import-position)",,,
"assignment/notebooks/dq_lib.py:11:0: C0413: Import ""import pytz"" should be placed at the top of the module (wrong-import-position)",,,,
assignment/notebooks/dq_lib.py:12:0: W0404: Reimport 'datetime' (imported line 1) (reimported),,,,
"assignment/notebooks/dq_lib.py:12:0: C0413: Import ""import datetime"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/notebooks/dq_lib.py:13:0: C0413: Import ""from collections import defaultdict"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/notebooks/dq_lib.py:15:0: C0413: Import ""import os"" should be placed at the top of the module (wrong-import-position)",,,,
assignment/notebooks/dq_lib.py:16:0: E0401: Unable to import 'findspark' (import-error),,,,
"assignment/notebooks/dq_lib.py:16:0: C0413: Import ""import findspark"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/notebooks/dq_lib.py:17:0: C0413: Import ""import time"" should be placed at the top of the module (wrong-import-position)",,,,
assignment/notebooks/dq_lib.py:21:0: E0401: Unable to import 'pyspark' (import-error),,,,
"assignment/notebooks/dq_lib.py:21:0: C0413: Import ""import pyspark"" should be placed at the top of the module (wrong-import-position)",,,,
assignment/notebooks/dq_lib.py:22:0: E0401: Unable to import 'pyspark.sql.types' (import-error),,,,
"assignment/notebooks/dq_lib.py:22:0: C0413: Import ""import pyspark.sql.types as sqlt"" should be placed at the top of the module (wrong-import-position)",,,,
assignment/notebooks/dq_lib.py:23:0: E0401: Unable to import 'pyspark.sql.functions' (import-error),,,,
"assignment/notebooks/dq_lib.py:23:0: C0413: Import ""import pyspark.sql.functions as sqlf"" should be placed at the top of the module (wrong-import-position)",,,,
assignment/notebooks/dq_lib.py:24:0: E0401: Unable to import 'pyspark.sql' (import-error),,,,
"assignment/notebooks/dq_lib.py:24:0: C0413: Import ""from pyspark.sql import Window as sqlw"" should be placed at the top of the module (wrong-import-position)",,,,
assignment/notebooks/dq_lib.py:25:0: E0401: Unable to import 'pyspark.sql.dataframe' (import-error),,,,
"assignment/notebooks/dq_lib.py:25:0: C0413: Import ""from pyspark.sql.dataframe import DataFrame"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/notebooks/dq_lib.py:27:0: C0413: Import ""import sys"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/notebooks/dq_lib.py:28:0: C0413: Import ""import re"" should be placed at the top of the module (wrong-import-position)",,,,
assignment/notebooks/dq_lib.py:35:0: C0116: Missing function or method docstring (missing-function-docstring),,,,
assignment/notebooks/dq_lib.py:36:4: W0603: Using the global statement (global-statement),,,,
assignment/notebooks/dq_lib.py:43:0: C0116: Missing function or method docstring (missing-function-docstring),,,,
assignment/notebooks/dq_lib.py:47:0: C0116: Missing function or method docstring (missing-function-docstring),,,,
assignment/notebooks/dq_lib.py:51:0: C0116: Missing function or method docstring (missing-function-docstring),,,,
assignment/notebooks/dq_lib.py:55:0: E0401: Unable to import 'IPython.core.display' (import-error),,,,
assignment/notebooks/dq_lib.py:55:0: W0404: Reimport 'display' (imported line 10) (reimported),,,,
assignment/notebooks/dq_lib.py:55:0: W0404: Reimport 'HTML' (imported line 10) (reimported),,,,
"assignment/notebooks/dq_lib.py:55:0: C0413: Import ""from IPython.core.display import display"," HTML"" should be placed at the top of the module (wrong-import-position)",,,
"assignment/notebooks/dq_lib.py:60:16: C0103: Argument name ""TICKET"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/notebooks/dq_lib.py:60:24: C0103: Argument name ""CONFIG"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/notebooks/dq_lib.py:60:0: C0116: Missing function or method docstring (missing-function-docstring),,,,
assignment/notebooks/dq_lib.py:61:4: C0415: Import outside toplevel (marty.lib.job_manager.JobManager) (import-outside-toplevel),,,,
assignment/notebooks/dq_lib.py:62:4: E0401: Unable to import 'pyspark.sql' (import-error),,,,
assignment/notebooks/dq_lib.py:62:4: C0415: Import outside toplevel (pyspark.sql.SparkSession) (import-outside-toplevel),,,,
assignment/notebooks/dq_lib.py:69:31: W0212: Access to a protected member _jsc of a client class (protected-access),,,,
"assignment/notebooks/dq_lib.py:76:8: C0103: Argument name ""n"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/notebooks/dq_lib.py:76:0: C0116: Missing function or method docstring (missing-function-docstring),,,,
assignment/notebooks/dq_lib.py:87:15: C0209: Formatting a regular string which could be a f-string (consider-using-f-string),,,,
"assignment/notebooks/dq_lib.py:93:9: C0103: Argument name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/notebooks/dq_lib.py:93:0: C0116: Missing function or method docstring (missing-function-docstring),,,,
assignment/notebooks/dq_lib.py:93:0: R1710: Either all return statements in a function should return an expression, or none of them should. (inconsistent-return-statements),,,
"assignment/notebooks/dq_lib.py:108:11: C0103: Argument name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/notebooks/dq_lib.py:108:0: C0116: Missing function or method docstring (missing-function-docstring),,,,
"assignment/notebooks/dq_lib.py:116:13: C0103: Argument name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/notebooks/dq_lib.py:116:0: C0116: Missing function or method docstring (missing-function-docstring),,,,
"assignment/notebooks/dq_lib.py:120:18: C0103: Argument name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/notebooks/dq_lib.py:120:0: C0116: Missing function or method docstring (missing-function-docstring),,,,
"assignment/notebooks/dq_lib.py:136:14: C0103: Argument name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/notebooks/dq_lib.py:136:0: C0116: Missing function or method docstring (missing-function-docstring),,,,
"assignment/notebooks/dq_lib.py:151:16: C0103: Argument name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/notebooks/dq_lib.py:151:0: C0116: Missing function or method docstring (missing-function-docstring),,,,
"assignment/notebooks/dq_lib.py:176:16: C0103: Argument name ""a"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/notebooks/dq_lib.py:176:19: C0103: Argument name ""b"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/notebooks/dq_lib.py:176:30: C0103: Argument name ""pk"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/notebooks/dq_lib.py:176:0: C0116: Missing function or method docstring (missing-function-docstring),,,,
"assignment/notebooks/dq_lib.py:191:23: C0103: Argument name ""a"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/notebooks/dq_lib.py:191:26: C0103: Argument name ""b"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/notebooks/dq_lib.py:191:29: C0103: Argument name ""pk"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/notebooks/dq_lib.py:191:0: C0116: Missing function or method docstring (missing-function-docstring),,,,
assignment/notebooks/dq_lib.py:205:19: W1114: Positional arguments appear to be out of order (arguments-out-of-order),,,,
assignment/notebooks/dq_lib.py:236:0: C0116: Missing function or method docstring (missing-function-docstring),,,,
"assignment/notebooks/dq_lib.py:237:4: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/notebooks/dq_lib.py:241:0: C0116: Missing function or method docstring (missing-function-docstring),,,,
"assignment/notebooks/dq_lib.py:242:4: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/notebooks/dq_lib.py:246:0: C0116: Missing function or method docstring (missing-function-docstring),,,,
assignment/notebooks/dq_lib.py:250:0: C0116: Missing function or method docstring (missing-function-docstring),,,,
assignment/notebooks/dq_lib.py:254:0: C0116: Missing function or method docstring (missing-function-docstring),,,,
assignment/notebooks/dq_lib.py:259:4: W0621: Redefining name 'get_path' from outer scope (line 357) (redefined-outer-name),,,,
assignment/notebooks/dq_lib.py:258:0: C0116: Missing function or method docstring (missing-function-docstring),,,,
assignment/notebooks/dq_lib.py:258:0: R0914: Too many local variables (24/15) (too-many-locals),,,,
assignment/notebooks/dq_lib.py:263:49: C0123: Use isinstance() rather than type() for a typecheck. (unidiomatic-typecheck),,,,
"assignment/notebooks/dq_lib.py:267:4: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/notebooks/dq_lib.py:276:8: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/notebooks/dq_lib.py:279:8: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/notebooks/dq_lib.py:281:8: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/notebooks/dq_lib.py:292:8: C0103: Variable name ""c"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/notebooks/dq_lib.py:294:8: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/notebooks/dq_lib.py:295:8: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/notebooks/dq_lib.py:307:56: C0209: Formatting a regular string which could be a f-string (consider-using-f-string),,,,
"assignment/notebooks/dq_lib.py:315:24: C0103: Variable name ""a"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/notebooks/dq_lib.py:315:27: C0103: Variable name ""b"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/notebooks/dq_lib.py:330:8: C0103: Variable name ""c"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/notebooks/dq_lib.py:330:11: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/notebooks/dq_lib.py:273:4: W0612: Unused variable 'missing_features_dates' (unused-variable),,,,
"assignment/notebooks/dq_lib.py:339:19: C0103: Argument name ""a"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/notebooks/dq_lib.py:339:22: C0103: Argument name ""b"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/notebooks/dq_lib.py:339:25: C0103: Argument name ""c"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/notebooks/dq_lib.py:339:0: C0116: Missing function or method docstring (missing-function-docstring),,,,
assignment/notebooks/dq_lib.py:352:0: E0102: function already defined line 236 (function-redefined),,,,
assignment/notebooks/dq_lib.py:352:0: C0116: Missing function or method docstring (missing-function-docstring),,,,
"assignment/notebooks/dq_lib.py:353:4: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/notebooks/dq_lib.py:357:0: C0116: Missing function or method docstring (missing-function-docstring),,,,
assignment/notebooks/dq_lib.py:364:0: C0116: Missing function or method docstring (missing-function-docstring),,,,
assignment/notebooks/dq_lib.py:371:0: W0102: Dangerous default value [] as argument (dangerous-default-value),,,,
assignment/notebooks/dq_lib.py:371:0: R0913: Too many arguments (8/5) (too-many-arguments),,,,
assignment/notebooks/dq_lib.py:371:0: R0914: Too many local variables (26/15) (too-many-locals),,,,
"assignment/notebooks/dq_lib.py:391:4: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/notebooks/dq_lib.py:403:8: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/notebooks/dq_lib.py:405:16: C0103: Variable name ""c"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/notebooks/dq_lib.py:407:16: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/notebooks/dq_lib.py:409:12: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/notebooks/dq_lib.py:417:8: W0702: No exception type(s) specified (bare-except),,,,
"assignment/notebooks/dq_lib.py:412:12: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/notebooks/dq_lib.py:414:12: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/notebooks/dq_lib.py:427:8: C0103: Variable name ""c"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/notebooks/dq_lib.py:429:8: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/notebooks/dq_lib.py:430:8: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/notebooks/dq_lib.py:442:56: C0209: Formatting a regular string which could be a f-string (consider-using-f-string),,,,
"assignment/notebooks/dq_lib.py:447:24: C0103: Variable name ""a"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/notebooks/dq_lib.py:447:27: C0103: Variable name ""b"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/notebooks/dq_lib.py:456:8: C0103: Variable name ""c"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/notebooks/dq_lib.py:456:11: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/notebooks/dq_lib.py:371:0: R0915: Too many statements (51/50) (too-many-statements),,,,
assignment/notebooks/dq_lib.py:15:0: W0611: Unused import os (unused-import),,,,
assignment/notebooks/dq_lib.py:21:0: W0611: Unused import pyspark (unused-import),,,,
assignment/notebooks/dq_lib.py:24:0: W0611: Unused Window imported from pyspark.sql as sqlw (unused-import),,,,
"assignment/notebooks/dq_lib.py:12:0: C0411: standard import ""import datetime"" should be placed before ""import pandas as pd"" (wrong-import-order)",,,,
"assignment/notebooks/dq_lib.py:13:0: C0411: standard import ""from collections import defaultdict"" should be placed before ""import pandas as pd"" (wrong-import-order)",,,,
"assignment/notebooks/dq_lib.py:15:0: C0411: standard import ""import os"" should be placed before ""import pandas as pd"" (wrong-import-order)",,,,
"assignment/notebooks/dq_lib.py:17:0: C0411: standard import ""import time"" should be placed before ""import pandas as pd"" (wrong-import-order)",,,,
"assignment/notebooks/dq_lib.py:27:0: C0411: standard import ""import sys"" should be placed before ""import pandas as pd"" (wrong-import-order)",,,,
"assignment/notebooks/dq_lib.py:28:0: C0411: standard import ""import re"" should be placed before ""import pandas as pd"" (wrong-import-order)",,,,
assignment/notebooks/dq_lib.py:12:0: C0412: Imports from package datetime are not grouped (ungrouped-imports),,,,
assignment/notebooks/dq_lib.py:55:0: C0412: Imports from package IPython are not grouped (ungrouped-imports),,,,
************* Module optimizer.lib.precheck,,,,
"assignment/optimizer/lib/precheck.py:11:4: C0103: Argument name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/optimizer/lib/precheck.py:12:4: C0103: Argument name ""Idx_CR"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/optimizer/lib/precheck.py:10:0: R0914: Too many local variables (16/15) (too-many-locals),,,,
"assignment/optimizer/lib/precheck.py:56:12: C0103: Variable name ""nm"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/optimizer/lib/precheck.py:75:20: C0209: Formatting a regular string which could be a f-string (consider-using-f-string),,,,
assignment/optimizer/lib/precheck.py:10:0: R0912: Too many branches (22/12) (too-many-branches),,,,
************* Module optimizer.lib.save_to_df,,,,
"assignment/optimizer/lib/save_to_df.py:10:23: C0103: Argument name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/optimizer/lib/save_to_df.py:67:12: C0209: Formatting a regular string which could be a f-string (consider-using-f-string),,,,
assignment/optimizer/lib/save_to_df.py:68:12: C0209: Formatting a regular string which could be a f-string (consider-using-f-string),,,,
************* Module optimizer.lib.summary,,,,
"assignment/optimizer/lib/summary.py:8:12: C0103: Argument name ""c1"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/optimizer/lib/summary.py:8:16: C0103: Argument name ""c2"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/optimizer/lib/summary.py:8:20: C0103: Argument name ""c3"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/optimizer/lib/summary.py:8:0: R0913: Too many arguments (7/5) (too-many-arguments),,,,
"assignment/optimizer/lib/summary.py:48:36: C0103: Argument name ""x0"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/optimizer/lib/summary.py:48:40: C0103: Argument name ""y0"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/optimizer/lib/summary.py:48:44: C0103: Argument name ""x1"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/optimizer/lib/summary.py:48:48: C0103: Argument name ""y1"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/optimizer/lib/summary.py:62:4: C0103: Variable name ""y2"" doesn't conform to snake_case naming style (invalid-name)",,,,
************* Module optimizer.lib.utils,,,,
assignment/optimizer/lib/utils.py:6:0: E0611: No name 'GRB' in module 'gurobipy' (no-name-in-module),,,,
assignment/optimizer/lib/utils.py:6:0: E0611: No name 'Model' in module 'gurobipy' (no-name-in-module),,,,
assignment/optimizer/lib/utils.py:6:0: E0611: No name 'quicksum' in module 'gurobipy' (no-name-in-module),,,,
assignment/optimizer/lib/utils.py:6:0: E0611: No name 'tupledict' in module 'gurobipy' (no-name-in-module),,,,
"assignment/optimizer/lib/utils.py:18:4: C0103: Argument name ""Idx_CR"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/optimizer/lib/utils.py:19:4: C0103: Argument name ""Idx_Clu"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/optimizer/lib/utils.py:21:4: C0103: Argument name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/optimizer/lib/utils.py:15:0: R0913: Too many arguments (9/5) (too-many-arguments),,,,
assignment/optimizer/lib/utils.py:15:0: R0914: Too many local variables (30/15) (too-many-locals),,,,
"assignment/optimizer/lib/utils.py:46:4: C0103: Variable name ""c1"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/optimizer/lib/utils.py:47:4: C0103: Variable name ""c2"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/optimizer/lib/utils.py:48:4: C0103: Variable name ""c3"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/optimizer/lib/utils.py:63:10: R1728: Consider using a generator instead 'tuple(int(y) for y in x.VarName[2:-1].split(','))' (consider-using-generator),,,
assignment/optimizer/lib/utils.py:85:8: R1722: Consider using sys.exit() (consider-using-sys-exit),,,,
assignment/optimizer/lib/utils.py:95:12: W0104: Statement seems to have no effect (pointless-statement),,,,
assignment/optimizer/lib/utils.py:96:12: W0104: Statement seems to have no effect (pointless-statement),,,,
assignment/optimizer/lib/utils.py:98:14: C0209: Formatting a regular string which could be a f-string (consider-using-f-string),,,,
assignment/optimizer/lib/utils.py:106:12: R1716: Simplify chained comparison between the operands (chained-comparison),,,,
assignment/optimizer/lib/utils.py:5:0: W0611: Unused numpy imported as np (unused-import),,,,
assignment/optimizer/lib/utils.py:6:0: W0611: Unused GRB imported from gurobipy (unused-import),,,,
assignment/optimizer/lib/utils.py:6:0: W0611: Unused Model imported from gurobipy (unused-import),,,,
************* Module optimizer.scripts.gurobi_optimizer_clusters,,,,
assignment/optimizer/scripts/gurobi_optimizer_clusters.py:293:0: C0301: Line too long (107/100) (line-too-long),,,,
assignment/optimizer/scripts/gurobi_optimizer_clusters.py:298:0: C0301: Line too long (107/100) (line-too-long),,,,
assignment/optimizer/scripts/gurobi_optimizer_clusters.py:12:0: E0611: No name 'GRB' in module 'gurobipy' (no-name-in-module),,,,
assignment/optimizer/scripts/gurobi_optimizer_clusters.py:12:0: E0611: No name 'Model' in module 'gurobipy' (no-name-in-module),,,,
assignment/optimizer/scripts/gurobi_optimizer_clusters.py:12:0: E0611: No name 'quicksum' in module 'gurobipy' (no-name-in-module),,,,
assignment/optimizer/scripts/gurobi_optimizer_clusters.py:12:0: E0611: No name 'tupledict' in module 'gurobipy' (no-name-in-module),,,,
assignment/optimizer/scripts/gurobi_optimizer_clusters.py:21:9: W0621: Redefining name 'cfg' from outer scope (line 338) (redefined-outer-name),,,,
assignment/optimizer/scripts/gurobi_optimizer_clusters.py:21:0: R0914: Too many local variables (53/15) (too-many-locals),,,,
"assignment/optimizer/scripts/gurobi_optimizer_clusters.py:34:8: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/optimizer/scripts/gurobi_optimizer_clusters.py:38:8: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/optimizer/scripts/gurobi_optimizer_clusters.py:65:4: C0103: Variable name ""c1"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/optimizer/scripts/gurobi_optimizer_clusters.py:66:4: C0103: Variable name ""c2"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/optimizer/scripts/gurobi_optimizer_clusters.py:67:4: C0103: Variable name ""c3"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/optimizer/scripts/gurobi_optimizer_clusters.py:76:7: R1729: Use a generator instead 'all(item in df.columns for item in count_cols + score_cols)' (use-a-generator),,,,
assignment/optimizer/scripts/gurobi_optimizer_clusters.py:86:7: C0121: Comparison 'cfg['main']['targeted_clubs'][0] != None' should be 'cfg['main']['targeted_clubs'][0] is not None' (singleton-comparison),,,,
assignment/optimizer/scripts/gurobi_optimizer_clusters.py:113:24: C0209: Formatting a regular string which could be a f-string (consider-using-f-string),,,,
"assignment/optimizer/scripts/gurobi_optimizer_clusters.py:119:4: C0103: Variable name ""Idx_Clu"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/optimizer/scripts/gurobi_optimizer_clusters.py:119:15: R1721: Unnecessary use of a comprehension, use list(range(no_clusters)) instead. (unnecessary-comprehension),,,
"assignment/optimizer/scripts/gurobi_optimizer_clusters.py:120:4: C0103: Variable name ""Idx_CR"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/optimizer/scripts/gurobi_optimizer_clusters.py:122:4: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/optimizer/scripts/gurobi_optimizer_clusters.py:135:4: C0103: Variable name ""startTime"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/optimizer/scripts/gurobi_optimizer_clusters.py:164:4: C0103: Variable name ""startTime2"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/optimizer/scripts/gurobi_optimizer_clusters.py:217:12: C0103: Variable name ""q"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/optimizer/scripts/gurobi_optimizer_clusters.py:218:12: C0103: Variable name ""A"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/optimizer/scripts/gurobi_optimizer_clusters.py:31:4: W0612: Unused variable 'output_dir' (unused-variable),,,,
assignment/optimizer/scripts/gurobi_optimizer_clusters.py:65:4: W0612: Unused variable 'c1' (unused-variable),,,,
assignment/optimizer/scripts/gurobi_optimizer_clusters.py:66:4: W0612: Unused variable 'c2' (unused-variable),,,,
assignment/optimizer/scripts/gurobi_optimizer_clusters.py:67:4: W0612: Unused variable 'c3' (unused-variable),,,,
assignment/optimizer/scripts/gurobi_optimizer_clusters.py:260:16: W0612: Unused variable 'total_cost' (unused-variable),,,,
assignment/optimizer/scripts/gurobi_optimizer_clusters.py:21:0: R0912: Too many branches (36/12) (too-many-branches),,,,
assignment/optimizer/scripts/gurobi_optimizer_clusters.py:21:0: R0915: Too many statements (135/50) (too-many-statements),,,,
assignment/optimizer/scripts/gurobi_optimizer_clusters.py:337:9: W1514: Using open without explicitly specifying an encoding (unspecified-encoding),,,,
assignment/optimizer/scripts/gurobi_optimizer_clusters.py:12:0: W0611: Unused tupledict imported from gurobipy (unused-import),,,,
************* Module scripts.check_campaign_master_table,,,,
"assignment/scripts/check_campaign_master_table.py:11:0: C0413: Import ""from marty.assignment.lib.campaign_master_table_validator import run_validation"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/check_campaign_master_table.py:12:0: C0413: Import ""from marty.common.python.io.dynamo import dynamo_read_table"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/check_campaign_master_table.py:13:0: C0413: Import ""from marty.lib.config_manager import ConfigManager"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/check_campaign_master_table.py:14:0: C0413: Import ""from marty.lib.job_manager import JobManager"" should be placed at the top of the module (wrong-import-position)",,,,
assignment/scripts/check_campaign_master_table.py:17:9: W0621: Redefining name 'job' from outer scope (line 76) (redefined-outer-name),,,,
************* Module scripts.check_input,,,,
assignment/scripts/check_input.py:14:9: W0621: Redefining name 'job' from outer scope (line 63) (redefined-outer-name),,,,
************* Module scripts.dq_trends,,,,
assignment/scripts/dq_trends.py:297:0: C0301: Line too long (106/100) (line-too-long),,,,
assignment/scripts/dq_trends.py:410:0: C0301: Line too long (196/100) (line-too-long),,,,
assignment/scripts/dq_trends.py:417:0: C0301: Line too long (271/100) (line-too-long),,,,
assignment/scripts/dq_trends.py:424:0: C0301: Line too long (210/100) (line-too-long),,,,
assignment/scripts/dq_trends.py:431:0: C0301: Line too long (286/100) (line-too-long),,,,
"assignment/scripts/dq_trends.py:16:0: C0413: Import ""import matplotlib.pyplot as plt"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/dq_trends.py:17:0: C0413: Import ""import pandas as pd"" should be placed at the top of the module (wrong-import-position)",,,,
assignment/scripts/dq_trends.py:18:0: E0401: Unable to import 'pyspark.sql.functions' (import-error),,,,
"assignment/scripts/dq_trends.py:18:0: C0413: Import ""import pyspark.sql.functions as sqlf"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/dq_trends.py:20:0: C0413: Import ""import marty.assignment.lib.utils as assignment_utils"" should be placed at the top of the module (wrong-import-position)",,,,
assignment/scripts/dq_trends.py:21:0: R0402: Use 'from marty.common.python.io import excel_report' instead (consider-using-from-import),,,,
"assignment/scripts/dq_trends.py:21:0: C0413: Import ""import marty.common.python.io.excel_report as excel_report"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/dq_trends.py:22:0: C0413: Import ""from marty.common.python.data.dq_checks import DqChecksExecutor"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/dq_trends.py:23:0: C0413: Import ""from marty.lib.config_manager import ConfigManager"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/dq_trends.py:24:0: C0413: Import ""from marty.lib.job_manager import JobManager"" should be placed at the top of the module (wrong-import-position)",,,,
assignment/scripts/dq_trends.py:29:25: W0621: Redefining name 'job' from outer scope (line 521) (redefined-outer-name),,,,
"assignment/scripts/dq_trends.py:42:13: C0103: Argument name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/scripts/dq_trends.py:42:17: C0103: Argument name ""q"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/scripts/dq_trends.py:88:13: W0621: Redefining name 'job' from outer scope (line 521) (redefined-outer-name),,,,
"assignment/scripts/dq_trends.py:105:4: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/scripts/dq_trends.py:109:23: W0621: Redefining name 'job' from outer scope (line 521) (redefined-outer-name),,,,
assignment/scripts/dq_trends.py:109:0: R0914: Too many local variables (22/15) (too-many-locals),,,,
"assignment/scripts/dq_trends.py:148:20: C0103: Variable name ""dq"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/scripts/dq_trends.py:159:8: R1714: Consider merging these comparisons with ""in"" to ""table in ('preselected_population_recent'"," 'preselected_population_intermediate_utuc_recent')"" (consider-using-in)",,,
assignment/scripts/dq_trends.py:173:4: C0206: Consider iterating with .items() (consider-using-dict-items),,,,
assignment/scripts/dq_trends.py:173:19: C0201: Consider iterating the dictionary directly instead of calling .keys() (consider-iterating-dictionary),,,,
"assignment/scripts/dq_trends.py:176:12: C0103: Variable name ""n"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/scripts/dq_trends.py:176:15: C0103: Variable name ""q"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/scripts/dq_trends.py:181:30: W0212: Access to a protected member _replace_parameters of a client class (protected-access),,,,
"assignment/scripts/dq_trends.py:188:16: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/scripts/dq_trends.py:198:16: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/scripts/dq_trends.py:156:4: W0612: Unused variable 'date_of_run' (unused-variable),,,,
assignment/scripts/dq_trends.py:210:27: W0621: Redefining name 'job' from outer scope (line 521) (redefined-outer-name),,,,
"assignment/scripts/dq_trends.py:235:4: W0106: Expression ""[dq_tables.pop(key) for key in list_of_paths_to_be_removed]"" is assigned to nothing (expression-not-assigned)",,,,
assignment/scripts/dq_trends.py:256:-1: W0105: String statement has no effect (pointless-string-statement),,,,
assignment/scripts/dq_trends.py:270:21: W0621: Redefining name 'job' from outer scope (line 521) (redefined-outer-name),,,,
assignment/scripts/dq_trends.py:270:0: R0914: Too many local variables (23/15) (too-many-locals),,,,
assignment/scripts/dq_trends.py:286:49: C0123: Use isinstance() rather than type() for a typecheck. (unidiomatic-typecheck),,,,
"assignment/scripts/dq_trends.py:290:4: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/scripts/dq_trends.py:296:4: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/scripts/dq_trends.py:308:8: C0103: Variable name ""c"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/scripts/dq_trends.py:310:8: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/scripts/dq_trends.py:311:8: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/scripts/dq_trends.py:323:56: C0209: Formatting a regular string which could be a f-string (consider-using-f-string),,,,
"assignment/scripts/dq_trends.py:329:24: C0103: Variable name ""a"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/scripts/dq_trends.py:329:27: C0103: Variable name ""b"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/scripts/dq_trends.py:342:8: C0103: Variable name ""c"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/scripts/dq_trends.py:342:11: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/scripts/dq_trends.py:358:22: W0621: Redefining name 'job' from outer scope (line 521) (redefined-outer-name),,,,
assignment/scripts/dq_trends.py:379:18: W0212: Access to a protected member _replace_parameters of a client class (protected-access),,,,
assignment/scripts/dq_trends.py:386:14: W0621: Redefining name 'job' from outer scope (line 521) (redefined-outer-name),,,,
assignment/scripts/dq_trends.py:436:4: W0621: Redefining name 'output_utuc' from outer scope (line 526) (redefined-outer-name),,,,
assignment/scripts/dq_trends.py:403:4: W1203: Use lazy % formatting in logging functions (logging-fstring-interpolation),,,,
assignment/scripts/dq_trends.py:440:19: W0621: Redefining name 'job' from outer scope (line 521) (redefined-outer-name),,,,
assignment/scripts/dq_trends.py:461:29: W0621: Redefining name 'output_utuc' from outer scope (line 526) (redefined-outer-name),,,,
"assignment/scripts/dq_trends.py:16:0: C0411: third party import ""import matplotlib.pyplot as plt"" should be placed before ""from marty.common.python.io.logger import MartyLogger"" (wrong-import-order)",,,,
"assignment/scripts/dq_trends.py:17:0: C0411: third party import ""import pandas as pd"" should be placed before ""from marty.common.python.io.logger import MartyLogger"" (wrong-import-order)",,,,
"assignment/scripts/dq_trends.py:18:0: C0411: third party import ""import pyspark.sql.functions as sqlf"" should be placed before ""from marty.common.python.io.logger import MartyLogger"" (wrong-import-order)",,,,
assignment/scripts/dq_trends.py:16:0: C0412: Imports from package matplotlib are not grouped (ungrouped-imports),,,,
************* Module scripts.dq_trends_final_assignment,,,,
assignment/scripts/dq_trends_final_assignment.py:283:0: C0301: Line too long (106/100) (line-too-long),,,,
"assignment/scripts/dq_trends_final_assignment.py:14:0: C0413: Import ""import matplotlib.pyplot as plt"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/dq_trends_final_assignment.py:15:0: C0413: Import ""import pandas as pd"" should be placed at the top of the module (wrong-import-position)",,,,
assignment/scripts/dq_trends_final_assignment.py:16:0: E0401: Unable to import 'pyspark.sql.functions' (import-error),,,,
"assignment/scripts/dq_trends_final_assignment.py:16:0: C0413: Import ""import pyspark.sql.functions as sqlf"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/dq_trends_final_assignment.py:18:0: C0413: Import ""import marty.assignment.lib.utils as assignment_utils"" should be placed at the top of the module (wrong-import-position)",,,,
assignment/scripts/dq_trends_final_assignment.py:19:0: R0402: Use 'from marty.common.python.io import excel_report' instead (consider-using-from-import),,,,
"assignment/scripts/dq_trends_final_assignment.py:19:0: C0413: Import ""import marty.common.python.io.excel_report as excel_report"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/dq_trends_final_assignment.py:20:0: C0413: Import ""from marty.common.python.data.dq_checks import DqChecksExecutor"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/dq_trends_final_assignment.py:21:0: C0413: Import ""from marty.lib.job_manager import JobManager"" should be placed at the top of the module (wrong-import-position)",,,,
assignment/scripts/dq_trends_final_assignment.py:26:25: W0621: Redefining name 'job' from outer scope (line 393) (redefined-outer-name),,,,
"assignment/scripts/dq_trends_final_assignment.py:46:13: C0103: Argument name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/scripts/dq_trends_final_assignment.py:46:17: C0103: Argument name ""q"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/scripts/dq_trends_final_assignment.py:92:13: W0621: Redefining name 'job' from outer scope (line 393) (redefined-outer-name),,,,
"assignment/scripts/dq_trends_final_assignment.py:109:4: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/scripts/dq_trends_final_assignment.py:113:23: W0621: Redefining name 'job' from outer scope (line 393) (redefined-outer-name),,,,
assignment/scripts/dq_trends_final_assignment.py:113:0: R0914: Too many local variables (23/15) (too-many-locals),,,,
"assignment/scripts/dq_trends_final_assignment.py:151:20: C0103: Variable name ""dq"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/scripts/dq_trends_final_assignment.py:176:4: C0206: Consider iterating with .items() (consider-using-dict-items),,,,
assignment/scripts/dq_trends_final_assignment.py:176:19: C0201: Consider iterating the dictionary directly instead of calling .keys() (consider-iterating-dictionary),,,,
"assignment/scripts/dq_trends_final_assignment.py:179:12: C0103: Variable name ""n"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/scripts/dq_trends_final_assignment.py:179:15: C0103: Variable name ""q"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/scripts/dq_trends_final_assignment.py:184:30: W0212: Access to a protected member _replace_parameters of a client class (protected-access),,,,
"assignment/scripts/dq_trends_final_assignment.py:191:16: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/scripts/dq_trends_final_assignment.py:201:16: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/scripts/dq_trends_final_assignment.py:159:4: W0612: Unused variable 'date_of_run' (unused-variable),,,,
assignment/scripts/dq_trends_final_assignment.py:213:27: W0621: Redefining name 'job' from outer scope (line 393) (redefined-outer-name),,,,
assignment/scripts/dq_trends_final_assignment.py:241:-1: W0105: String statement has no effect (pointless-string-statement),,,,
assignment/scripts/dq_trends_final_assignment.py:256:21: W0621: Redefining name 'job' from outer scope (line 393) (redefined-outer-name),,,,
assignment/scripts/dq_trends_final_assignment.py:256:0: R0914: Too many local variables (23/15) (too-many-locals),,,,
assignment/scripts/dq_trends_final_assignment.py:272:49: C0123: Use isinstance() rather than type() for a typecheck. (unidiomatic-typecheck),,,,
"assignment/scripts/dq_trends_final_assignment.py:276:4: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/scripts/dq_trends_final_assignment.py:282:4: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/scripts/dq_trends_final_assignment.py:294:8: C0103: Variable name ""c"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/scripts/dq_trends_final_assignment.py:296:8: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/scripts/dq_trends_final_assignment.py:297:8: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/scripts/dq_trends_final_assignment.py:309:56: C0209: Formatting a regular string which could be a f-string (consider-using-f-string),,,,
"assignment/scripts/dq_trends_final_assignment.py:315:24: C0103: Variable name ""a"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/scripts/dq_trends_final_assignment.py:315:27: C0103: Variable name ""b"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/scripts/dq_trends_final_assignment.py:328:8: C0103: Variable name ""c"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/scripts/dq_trends_final_assignment.py:328:11: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/scripts/dq_trends_final_assignment.py:344:19: W0621: Redefining name 'job' from outer scope (line 393) (redefined-outer-name),,,,
assignment/scripts/dq_trends_final_assignment.py:9:0: W0611: Unused lru_cache imported from functools (unused-import),,,,
assignment/scripts/dq_trends_final_assignment.py:16:0: W0611: Unused pyspark.sql.functions imported as sqlf (unused-import),,,,
assignment/scripts/dq_trends_final_assignment.py:18:0: W0611: Unused marty.assignment.lib.utils imported as assignment_utils (unused-import),,,,
************* Module scripts.etl_biz_bronze_pros,,,,
assignment/scripts/etl_biz_bronze_pros.py:11:0: E0401: Unable to import 'pyspark.storagelevel' (import-error),,,,
"assignment/scripts/etl_biz_bronze_pros.py:11:0: C0413: Import ""from pyspark.storagelevel import StorageLevel"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/etl_biz_bronze_pros.py:13:0: C0413: Import ""from marty.common.python.io.s3 import input_data_validator"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/etl_biz_bronze_pros.py:14:0: C0413: Import ""from marty.lib.config_manager import ConfigManager"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/etl_biz_bronze_pros.py:15:0: C0413: Import ""from marty.lib.job_manager import JobManager"" should be placed at the top of the module (wrong-import-position)",,,,
assignment/scripts/etl_biz_bronze_pros.py:18:24: W0621: Redefining name 'job' from outer scope (line 107) (redefined-outer-name),,,,
"assignment/scripts/etl_biz_bronze_pros.py:49:8: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/scripts/etl_biz_bronze_pros.py:50:8: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/scripts/etl_biz_bronze_pros.py:51:8: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/scripts/etl_biz_bronze_pros.py:54:8: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/scripts/etl_biz_bronze_pros.py:11:0: C0411: third party import ""from pyspark.storagelevel import StorageLevel"" should be placed before ""from marty.common.python.io.logger import MartyLogger"" (wrong-import-order)",,,,
assignment/scripts/etl_biz_bronze_pros.py:13:0: C0412: Imports from package marty are not grouped (ungrouped-imports),,,,
************* Module scripts.etl_campaign_history,,,,
assignment/scripts/etl_campaign_history.py:71:0: C0301: Line too long (101/100) (line-too-long),,,,
assignment/scripts/etl_campaign_history.py:7:0: E0401: Unable to import 'pyspark.sql.functions' (import-error),,,,
assignment/scripts/etl_campaign_history.py:15:17: W0621: Redefining name 'job' from outer scope (line 115) (redefined-outer-name),,,,
************* Module scripts.file_transfer_assignment,,,,
assignment/scripts/file_transfer_assignment.py:8:0: R0402: Use 'from marty.assignment.lib import file_transfer_utils' instead (consider-using-from-import),,,,
assignment/scripts/file_transfer_assignment.py:38:8: W0621: Redefining name 'job' from outer scope (line 102) (redefined-outer-name),,,,
************* Module scripts.file_transfer_final_assignment,,,,
assignment/scripts/file_transfer_final_assignment.py:8:0: R0402: Use 'from marty.assignment.lib import file_transfer_utils' instead (consider-using-from-import),,,,
assignment/scripts/file_transfer_final_assignment.py:35:8: W0621: Redefining name 'job' from outer scope (line 94) (redefined-outer-name),,,,
************* Module scripts.final_post_dq_checks,,,,
assignment/scripts/final_post_dq_checks.py:11:0: E0401: Unable to import 'pyspark.sql' (import-error),,,,
"assignment/scripts/final_post_dq_checks.py:11:0: C0413: Import ""from pyspark.sql import Row as sqlr"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/final_post_dq_checks.py:13:0: C0413: Import ""from marty.assignment.lib.final_dq_checks import FinalAssignmentDqChecksExecutor"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/final_post_dq_checks.py:16:0: C0413: Import ""from marty.common.python.io.dynamo import dynamo_read_table"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/final_post_dq_checks.py:17:0: C0413: Import ""from marty.common.python.io.s3 import input_data_validator"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/final_post_dq_checks.py:18:0: C0413: Import ""from marty.lib.config_manager import ConfigManager"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/final_post_dq_checks.py:19:0: C0413: Import ""from marty.lib.job_manager import JobManager"" should be placed at the top of the module (wrong-import-position)",,,,
assignment/scripts/final_post_dq_checks.py:22:19: W0621: Redefining name 'job' from outer scope (line 112) (redefined-outer-name),,,,
assignment/scripts/final_post_dq_checks.py:46:37: W0621: Redefining name 'job' from outer scope (line 112) (redefined-outer-name),,,,
assignment/scripts/final_post_dq_checks.py:11:0: W0611: Unused Row imported from pyspark.sql as sqlr (unused-import),,,,
assignment/scripts/final_post_dq_checks.py:16:0: W0611: Unused dynamo_read_table imported from marty.common.python.io.dynamo (unused-import),,,,
"assignment/scripts/final_post_dq_checks.py:11:0: C0411: third party import ""from pyspark.sql import Row as sqlr"" should be placed before ""from marty.common.python.io.logger import MartyLogger"" (wrong-import-order)",,,,
assignment/scripts/final_post_dq_checks.py:13:0: C0412: Imports from package marty are not grouped (ungrouped-imports),,,,
************* Module scripts.forced_cell_assignment,,,,
"assignment/scripts/forced_cell_assignment.py:11:0: C0413: Import ""from marty.assignment.lib.cell_assignment import forced_cell_assignment"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/forced_cell_assignment.py:12:0: C0413: Import ""from marty.assignment.lib.uplift_utils import apply_uplift_exclusions"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/forced_cell_assignment.py:13:0: C0413: Import ""from marty.common.python.io.dynamo import dynamo_read_table"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/forced_cell_assignment.py:14:0: C0413: Import ""from marty.common.python.io.s3 import input_data_validator"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/forced_cell_assignment.py:15:0: C0413: Import ""from marty.lib.config_manager import ConfigManager"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/forced_cell_assignment.py:16:0: C0413: Import ""from marty.lib.job_manager import JobManager"" should be placed at the top of the module (wrong-import-position)",,,,
assignment/scripts/forced_cell_assignment.py:19:22: W0621: Redefining name 'job' from outer scope (line 184) (redefined-outer-name),,,,
assignment/scripts/forced_cell_assignment.py:19:37: W0621: Redefining name 'allowed_paths' from outer scope (line 192) (redefined-outer-name),,,,
"assignment/scripts/forced_cell_assignment.py:19:27: C0103: Argument name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/scripts/forced_cell_assignment.py:42:8: W0621: Redefining name 'job' from outer scope (line 184) (redefined-outer-name),,,,
assignment/scripts/forced_cell_assignment.py:42:13: W0621: Redefining name 'allowed_paths' from outer scope (line 192) (redefined-outer-name),,,,
assignment/scripts/forced_cell_assignment.py:42:0: R0914: Too many local variables (17/15) (too-many-locals),,,,
************* Module scripts.generate_gurobi_input,,,,
"assignment/scripts/generate_gurobi_input.py:11:0: C0413: Import ""from marty.assignment.lib.gurobi_input_preparation import prepare_gurobi_input"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/generate_gurobi_input.py:12:0: C0413: Import ""from marty.common.python.io.s3 import input_data_validator"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/generate_gurobi_input.py:13:0: C0413: Import ""from marty.lib.config_manager import ConfigManager"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/generate_gurobi_input.py:14:0: C0413: Import ""from marty.lib.job_manager import JobManager"" should be placed at the top of the module (wrong-import-position)",,,,
assignment/scripts/generate_gurobi_input.py:17:22: W0621: Redefining name 'job' from outer scope (line 120) (redefined-outer-name),,,,
assignment/scripts/generate_gurobi_input.py:17:64: W0621: Redefining name 'allowed_paths' from outer scope (line 128) (redefined-outer-name),,,,
assignment/scripts/generate_gurobi_input.py:44:8: W0621: Redefining name 'job' from outer scope (line 120) (redefined-outer-name),,,,
assignment/scripts/generate_gurobi_input.py:44:13: W0621: Redefining name 'allowed_paths' from outer scope (line 128) (redefined-outer-name),,,,
************* Module scripts.gurobi_step,,,,
assignment/scripts/gurobi_step.py:14:8: W0621: Redefining name 'job' from outer scope (line 90) (redefined-outer-name),,,,
assignment/scripts/gurobi_step.py:14:13: W0621: Redefining name 'gurobi_config_path' from outer scope (line 92) (redefined-outer-name),,,,
************* Module scripts.main_cell_assignment,,,,
assignment/scripts/main_cell_assignment.py:11:0: E0401: Unable to import 'pyspark.sql.functions' (import-error),,,,
assignment/scripts/main_cell_assignment.py:12:0: E0401: Unable to import 'pyspark.sql.types' (import-error),,,,
"assignment/scripts/main_cell_assignment.py:17:0: C0413: Import ""import marty.common.python.io.s3 as io_s3"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/main_cell_assignment.py:18:0: C0413: Import ""import marty.common.python.spark.misc.utils as misc_utils"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/main_cell_assignment.py:19:0: C0413: Import ""from marty.assignment.lib.cell_assignment import main_cell_assignment"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/main_cell_assignment.py:20:0: C0413: Import ""from marty.assignment.lib.main_cell_eligible import tag_main_cell_eligible"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/main_cell_assignment.py:21:0: C0413: Import ""from marty.common.python.io.dynamo import dynamo_read_table"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/main_cell_assignment.py:22:0: C0413: Import ""from marty.common.python.io.s3 import input_data_validator"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/main_cell_assignment.py:23:0: C0413: Import ""from marty.lib.config_manager import ConfigManager"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/main_cell_assignment.py:24:0: C0413: Import ""from marty.lib.job_manager import JobManager"" should be placed at the top of the module (wrong-import-position)",,,,
assignment/scripts/main_cell_assignment.py:27:26: W0621: Redefining name 'job' from outer scope (line 453) (redefined-outer-name),,,,
assignment/scripts/main_cell_assignment.py:39:4: C0115: Missing class docstring (missing-class-docstring),,,,
assignment/scripts/main_cell_assignment.py:43:19: R1725: Consider using Python 3 style super() without arguments (super-with-arguments),,,,
assignment/scripts/main_cell_assignment.py:58:4: W0621: Redefining name 'job' from outer scope (line 453) (redefined-outer-name),,,,
assignment/scripts/main_cell_assignment.py:58:45: W0621: Redefining name 'allowed_paths' from outer scope (line 461) (redefined-outer-name),,,,
assignment/scripts/main_cell_assignment.py:57:0: R0913: Too many arguments (6/5) (too-many-arguments),,,,
assignment/scripts/main_cell_assignment.py:85:25: W0621: Redefining name 'job' from outer scope (line 453) (redefined-outer-name),,,,
assignment/scripts/main_cell_assignment.py:126:25: W0621: Redefining name 'job' from outer scope (line 453) (redefined-outer-name),,,,
assignment/scripts/main_cell_assignment.py:180:26: W0621: Redefining name 'job' from outer scope (line 453) (redefined-outer-name),,,,
assignment/scripts/main_cell_assignment.py:180:55: W0621: Redefining name 'main_cell_assignment' from outer scope (line 19) (redefined-outer-name),,,,
assignment/scripts/main_cell_assignment.py:245:8: W0621: Redefining name 'job' from outer scope (line 453) (redefined-outer-name),,,,
assignment/scripts/main_cell_assignment.py:245:13: W0621: Redefining name 'allowed_paths' from outer scope (line 461) (redefined-outer-name),,,,
assignment/scripts/main_cell_assignment.py:245:0: R0914: Too many local variables (20/15) (too-many-locals),,,,
************* Module scripts.post_dq_checks,,,,
assignment/scripts/post_dq_checks.py:12:0: E0401: Unable to import 'pyspark.sql.functions' (import-error),,,,
"assignment/scripts/post_dq_checks.py:12:0: C0413: Import ""import pyspark.sql.functions as sqlf"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/post_dq_checks.py:14:0: C0413: Import ""import marty.assignment.lib.utils as assignment_utils"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/post_dq_checks.py:15:0: C0413: Import ""from marty.common.python.data.dq_checks import DataQualityException"," DqChecksExecutor"" should be placed at the top of the module (wrong-import-position)",,,
"assignment/scripts/post_dq_checks.py:19:0: C0413: Import ""from marty.common.python.io.s3 import input_data_validator"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/post_dq_checks.py:20:0: C0413: Import ""from marty.lib.config_manager import ConfigManager"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/post_dq_checks.py:21:0: C0413: Import ""from marty.lib.job_manager import JobManager"" should be placed at the top of the module (wrong-import-position)",,,,
assignment/scripts/post_dq_checks.py:24:19: W0621: Redefining name 'job' from outer scope (line 282) (redefined-outer-name),,,,
"assignment/scripts/post_dq_checks.py:70:8: C0103: Variable name ""e"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/scripts/post_dq_checks.py:79:29: W0621: Redefining name 'job' from outer scope (line 282) (redefined-outer-name),,,,
assignment/scripts/post_dq_checks.py:105:30: W0621: Redefining name 'job' from outer scope (line 282) (redefined-outer-name),,,,
"assignment/scripts/post_dq_checks.py:148:8: C0103: Variable name ""t"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/scripts/post_dq_checks.py:167:12: C0103: Variable name ""t"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/scripts/post_dq_checks.py:179:23: W0621: Redefining name 'job' from outer scope (line 282) (redefined-outer-name),,,,
"assignment/scripts/post_dq_checks.py:229:4: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/scripts/post_dq_checks.py:12:0: C0411: third party import ""import pyspark.sql.functions as sqlf"" should be placed before ""from marty.common.python.io.logger import MartyLogger"" (wrong-import-order)",,,,
************* Module scripts.post_supp_dq_checks,,,,
assignment/scripts/post_supp_dq_checks.py:12:0: E0401: Unable to import 'pyspark.sql.functions' (import-error),,,,
"assignment/scripts/post_supp_dq_checks.py:12:0: C0413: Import ""import pyspark.sql.functions as sqlf"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/post_supp_dq_checks.py:14:0: C0413: Import ""import marty.assignment.lib.utils as assignment_utils"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/post_supp_dq_checks.py:15:0: C0413: Import ""from marty.common.python.data.dq_checks import DataQualityException"," DqChecksExecutor"" should be placed at the top of the module (wrong-import-position)",,,
"assignment/scripts/post_supp_dq_checks.py:19:0: C0413: Import ""from marty.lib.config_manager import ConfigManager"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/post_supp_dq_checks.py:20:0: C0413: Import ""from marty.lib.job_manager import JobManager"" should be placed at the top of the module (wrong-import-position)",,,,
assignment/scripts/post_supp_dq_checks.py:23:19: W0621: Redefining name 'job' from outer scope (line 232) (redefined-outer-name),,,,
"assignment/scripts/post_supp_dq_checks.py:69:8: C0103: Variable name ""e"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/scripts/post_supp_dq_checks.py:78:29: W0621: Redefining name 'job' from outer scope (line 232) (redefined-outer-name),,,,
assignment/scripts/post_supp_dq_checks.py:96:30: W0621: Redefining name 'job' from outer scope (line 232) (redefined-outer-name),,,,
"assignment/scripts/post_supp_dq_checks.py:128:8: C0103: Variable name ""t"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/scripts/post_supp_dq_checks.py:142:23: W0621: Redefining name 'job' from outer scope (line 232) (redefined-outer-name),,,,
"assignment/scripts/post_supp_dq_checks.py:179:4: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/scripts/post_supp_dq_checks.py:12:0: C0411: third party import ""import pyspark.sql.functions as sqlf"" should be placed before ""from marty.common.python.io.logger import MartyLogger"" (wrong-import-order)",,,,
************* Module scripts.pre_selection,,,,
assignment/scripts/pre_selection.py:11:0: E0401: Unable to import 'pyspark.ml.feature' (import-error),,,,
"assignment/scripts/pre_selection.py:11:0: C0413: Import ""import pyspark.ml.feature as mlfeat"" should be placed at the top of the module (wrong-import-position)",,,,
assignment/scripts/pre_selection.py:12:0: E0401: Unable to import 'pyspark.sql.functions' (import-error),,,,
"assignment/scripts/pre_selection.py:12:0: C0413: Import ""import pyspark.sql.functions as sqlf"" should be placed at the top of the module (wrong-import-position)",,,,
assignment/scripts/pre_selection.py:13:0: E0401: Unable to import 'pyspark.sql.types' (import-error),,,,
"assignment/scripts/pre_selection.py:13:0: C0413: Import ""import pyspark.sql.types as sqlt"" should be placed at the top of the module (wrong-import-position)",,,,
assignment/scripts/pre_selection.py:14:0: E0401: Unable to import 'pyspark.sql' (import-error),,,,
"assignment/scripts/pre_selection.py:14:0: C0413: Import ""from pyspark.sql import Window as sqlw"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/pre_selection.py:16:0: C0413: Import ""import marty.assignment.lib.utils as assignment_utils"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/pre_selection.py:17:0: C0413: Import ""from marty.common.python.io.s3 import input_data_validator"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/pre_selection.py:18:0: C0413: Import ""from marty.lib.config_manager import ConfigManager"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/pre_selection.py:19:0: C0413: Import ""from marty.lib.job_manager import JobManager"" should be placed at the top of the module (wrong-import-position)",,,,
assignment/scripts/pre_selection.py:257:4: W1203: Use lazy % formatting in logging functions (logging-fstring-interpolation),,,,
"assignment/scripts/pre_selection.py:337:16: C0103: Argument name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/scripts/pre_selection.py:360:33: W0621: Redefining name 'job' from outer scope (line 586) (redefined-outer-name),,,,
assignment/scripts/pre_selection.py:411:4: W0621: Redefining name 'job' from outer scope (line 586) (redefined-outer-name),,,,
assignment/scripts/pre_selection.py:411:27: W0621: Redefining name 'allowed_paths' from outer scope (line 594) (redefined-outer-name),,,,
assignment/scripts/pre_selection.py:437:4: W0621: Redefining name 'job' from outer scope (line 586) (redefined-outer-name),,,,
assignment/scripts/pre_selection.py:437:28: W0621: Redefining name 'allowed_paths' from outer scope (line 594) (redefined-outer-name),,,,
assignment/scripts/pre_selection.py:459:8: W0621: Redefining name 'job' from outer scope (line 586) (redefined-outer-name),,,,
assignment/scripts/pre_selection.py:459:13: W0621: Redefining name 'allowed_paths' from outer scope (line 594) (redefined-outer-name),,,,
"assignment/scripts/pre_selection.py:11:0: C0411: third party import ""import pyspark.ml.feature as mlfeat"" should be placed before ""from marty.common.python.io.logger import MartyLogger"" (wrong-import-order)",,,,
"assignment/scripts/pre_selection.py:12:0: C0411: third party import ""import pyspark.sql.functions as sqlf"" should be placed before ""from marty.common.python.io.logger import MartyLogger"" (wrong-import-order)",,,,
"assignment/scripts/pre_selection.py:13:0: C0411: third party import ""import pyspark.sql.types as sqlt"" should be placed before ""from marty.common.python.io.logger import MartyLogger"" (wrong-import-order)",,,,
"assignment/scripts/pre_selection.py:14:0: C0411: third party import ""from pyspark.sql import Window as sqlw"" should be placed before ""from marty.common.python.io.logger import MartyLogger"" (wrong-import-order)",,,,
************* Module scripts.pre_selection_intermediate,,,,
assignment/scripts/pre_selection_intermediate.py:12:0: E0401: Unable to import 'pyspark.sql.functions' (import-error),,,,
"assignment/scripts/pre_selection_intermediate.py:12:0: C0413: Import ""import pyspark.sql.functions as sqlf"" should be placed at the top of the module (wrong-import-position)",,,,
assignment/scripts/pre_selection_intermediate.py:14:0: R0402: Use 'from marty.assignment.lib import assignment_utuc' instead (consider-using-from-import),,,,
"assignment/scripts/pre_selection_intermediate.py:14:0: C0413: Import ""import marty.assignment.lib.assignment_utuc as assignment_utuc"" should be placed at the top of the module (wrong-import-position)",,,,
assignment/scripts/pre_selection_intermediate.py:15:0: R0402: Use 'from marty.assignment.lib import model_scores' instead (consider-using-from-import),,,,
"assignment/scripts/pre_selection_intermediate.py:15:0: C0413: Import ""import marty.assignment.lib.model_scores as model_scores"" should be placed at the top of the module (wrong-import-position)",,,,
assignment/scripts/pre_selection_intermediate.py:16:0: R0402: Use 'from marty.assignment.lib import offer_utils' instead (consider-using-from-import),,,,
"assignment/scripts/pre_selection_intermediate.py:16:0: C0413: Import ""import marty.assignment.lib.offer_utils as offer_utils"" should be placed at the top of the module (wrong-import-position)",,,,
assignment/scripts/pre_selection_intermediate.py:17:0: R0402: Use 'from marty.assignment.lib import pre_selection_utils' instead (consider-using-from-import),,,,
"assignment/scripts/pre_selection_intermediate.py:17:0: C0413: Import ""import marty.assignment.lib.pre_selection_utils as pre_selection_utils"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/pre_selection_intermediate.py:18:0: C0413: Import ""import marty.assignment.lib.utils as assignment_utils"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/pre_selection_intermediate.py:19:0: C0413: Import ""import marty.common.python.spark.misc.utils as misc_utils"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/pre_selection_intermediate.py:20:0: C0413: Import ""from marty.common.python.io.s3 import input_data_validator"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/pre_selection_intermediate.py:21:0: C0413: Import ""from marty.lib.config_manager import ConfigManager"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/pre_selection_intermediate.py:22:0: C0413: Import ""from marty.lib.job_manager import JobManager"" should be placed at the top of the module (wrong-import-position)",,,,
assignment/scripts/pre_selection_intermediate.py:25:18: W0621: Redefining name 'job' from outer scope (line 370) (redefined-outer-name),,,,
assignment/scripts/pre_selection_intermediate.py:25:23: W0621: Redefining name 'allowed_paths' from outer scope (line 366) (redefined-outer-name),,,,
assignment/scripts/pre_selection_intermediate.py:25:0: R0914: Too many local variables (49/15) (too-many-locals),,,,
assignment/scripts/pre_selection_intermediate.py:25:0: R0915: Too many statements (78/50) (too-many-statements),,,,
"assignment/scripts/pre_selection_intermediate.py:12:0: C0411: third party import ""import pyspark.sql.functions as sqlf"" should be placed before ""from marty.common.python.io.logger import MartyLogger"" (wrong-import-order)",,,,
************* Module scripts.pre_selection_suppression_files,,,,
assignment/scripts/pre_selection_suppression_files.py:14:0: E0401: Unable to import 'pyspark.sql' (import-error),,,,
"assignment/scripts/pre_selection_suppression_files.py:14:0: C0413: Import ""from pyspark.sql import functions as sqlf"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/pre_selection_suppression_files.py:16:0: C0413: Import ""import marty.assignment.lib.utils as assignment_utils"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/pre_selection_suppression_files.py:17:0: C0413: Import ""from marty.common.python.io.s3 import input_data_validator"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/pre_selection_suppression_files.py:18:0: C0413: Import ""from marty.lib.job_manager import JobManager"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/pre_selection_suppression_files.py:19:0: C0413: Import ""from marty.lib.schemas.initial_selection_suppression_file_schema import COLS_WIDTHS"" should be placed at the top of the module (wrong-import-position)",,,,
assignment/scripts/pre_selection_suppression_files.py:24:0: R0902: Too many instance attributes (8/7) (too-many-instance-attributes),,,,
assignment/scripts/pre_selection_suppression_files.py:40:33: W0621: Redefining name 'config_path' from outer scope (line 300) (redefined-outer-name),,,,
assignment/scripts/pre_selection_suppression_files.py:70:18: W0621: Redefining name 'allowed_paths' from outer scope (line 303) (redefined-outer-name),,,,
assignment/scripts/pre_selection_suppression_files.py:235:25: W0621: Redefining name 'allowed_paths' from outer scope (line 303) (redefined-outer-name),,,,
"assignment/scripts/pre_selection_suppression_files.py:14:0: C0411: third party import ""from pyspark.sql import functions as sqlf"" should be placed before ""from marty.common.python.io.logger import MartyLogger"" (wrong-import-order)",,,,
************* Module scripts.second_suppression_and_segmentation,,,,
assignment/scripts/second_suppression_and_segmentation.py:13:0: E0401: Unable to import 'pyspark.sql.functions' (import-error),,,,
"assignment/scripts/second_suppression_and_segmentation.py:13:0: C0413: Import ""import pyspark.sql.functions as sqlf"" should be placed at the top of the module (wrong-import-position)",,,,
assignment/scripts/second_suppression_and_segmentation.py:14:0: E0401: Unable to import 'pyspark.sql.types' (import-error),,,,
"assignment/scripts/second_suppression_and_segmentation.py:14:0: C0413: Import ""import pyspark.sql.types as sqlt"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/second_suppression_and_segmentation.py:16:0: C0413: Import ""import marty.assignment.lib.second_suppression_and_segmentation_utils as second_utils"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/second_suppression_and_segmentation.py:17:0: C0413: Import ""import marty.assignment.lib.utils as assignment_utils"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/second_suppression_and_segmentation.py:18:0: C0413: Import ""from marty.assignment.lib.pcoa_utils import add_pcoa"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/second_suppression_and_segmentation.py:19:0: C0413: Import ""from marty.common.python.io.s3 import input_data_validator"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/second_suppression_and_segmentation.py:20:0: C0413: Import ""from marty.lib.config_manager import ConfigManager"" should be placed at the top of the module (wrong-import-position)",,,,
"assignment/scripts/second_suppression_and_segmentation.py:21:0: C0413: Import ""from marty.lib.job_manager import JobManager"" should be placed at the top of the module (wrong-import-position)",,,,
assignment/scripts/second_suppression_and_segmentation.py:24:8: W0621: Redefining name 'job' from outer scope (line 226) (redefined-outer-name),,,,
assignment/scripts/second_suppression_and_segmentation.py:24:13: W0621: Redefining name 'allowed_paths' from outer scope (line 235) (redefined-outer-name),,,,
assignment/scripts/second_suppression_and_segmentation.py:24:0: R0914: Too many local variables (24/15) (too-many-locals),,,,
"assignment/scripts/second_suppression_and_segmentation.py:13:0: C0411: third party import ""import pyspark.sql.functions as sqlf"" should be placed before ""from marty.common.python.io.logger import MartyLogger"" (wrong-import-order)",,,,
"assignment/scripts/second_suppression_and_segmentation.py:14:0: C0411: third party import ""import pyspark.sql.types as sqlt"" should be placed before ""from marty.common.python.io.logger import MartyLogger"" (wrong-import-order)",,,,
************* Module tests.regression.assignment_regression,,,,
assignment/tests/regression/assignment_regression.py:124:0: C0301: Line too long (108/100) (line-too-long),,,,
assignment/tests/regression/assignment_regression.py:11:0: E0401: Unable to import 'pyspark.sql.functions' (import-error),,,,
assignment/tests/regression/assignment_regression.py:80:12: W0702: No exception type(s) specified (bare-except),,,,
"assignment/tests/regression/assignment_regression.py:101:30: C0103: Argument name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/tests/regression/assignment_regression.py:101:4: R0201: Method could be a function (no-self-use),,,,
"assignment/tests/regression/assignment_regression.py:128:8: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/tests/regression/assignment_regression.py:168:29: C0201: Consider iterating the dictionary directly instead of calling .keys() (consider-iterating-dictionary),,,,
assignment/tests/regression/assignment_regression.py:199:8: W0621: Redefining name 'forced_cell_assignment' from outer scope (line 19) (redefined-outer-name),,,,
assignment/tests/regression/assignment_regression.py:211:58: C0209: Formatting a regular string which could be a f-string (consider-using-f-string),,,,
************* Module tests.unit.campaign_master_table_validator_unit,,,,
assignment/tests/unit/campaign_master_table_validator_unit.py:101:4: R0201: Method could be a function (no-self-use),,,,
assignment/tests/unit/campaign_master_table_validator_unit.py:120:4: R0201: Method could be a function (no-self-use),,,,
assignment/tests/unit/campaign_master_table_validator_unit.py:137:4: R0201: Method could be a function (no-self-use),,,,
assignment/tests/unit/campaign_master_table_validator_unit.py:154:4: R0201: Method could be a function (no-self-use),,,,
assignment/tests/unit/campaign_master_table_validator_unit.py:187:45: E0602: Undefined variable 'invalid_campaign_master_table' (undefined-variable),,,,
assignment/tests/unit/campaign_master_table_validator_unit.py:176:8: W0612: Unused variable 'invalid_campaign_mater_table' (unused-variable),,,,
assignment/tests/unit/campaign_master_table_validator_unit.py:205:45: E0602: Undefined variable 'invalid_campaign_master_table' (undefined-variable),,,,
assignment/tests/unit/campaign_master_table_validator_unit.py:194:8: W0612: Unused variable 'invalid_campaign_mater_table' (unused-variable),,,,
assignment/tests/unit/campaign_master_table_validator_unit.py:223:45: E0602: Undefined variable 'invalid_campaign_master_table' (undefined-variable),,,,
assignment/tests/unit/campaign_master_table_validator_unit.py:212:8: W0612: Unused variable 'invalid_campaign_mater_table' (unused-variable),,,,
assignment/tests/unit/campaign_master_table_validator_unit.py:244:39: E0602: Undefined variable 'invalid_campaign_master_table' (undefined-variable),,,,
assignment/tests/unit/campaign_master_table_validator_unit.py:229:8: W0612: Unused variable 'campaign_mater_table' (unused-variable),,,,
assignment/tests/unit/campaign_master_table_validator_unit.py:246:4: E0102: method already defined line 225 (function-redefined),,,,
assignment/tests/unit/campaign_master_table_validator_unit.py:246:4: R0201: Method could be a function (no-self-use),,,,
************* Module tests.unit.check_input_unit,,,,
assignment/tests/unit/check_input_unit.py:42:0: C0301: Line too long (120/100) (line-too-long),,,,
assignment/tests/unit/check_input_unit.py:43:0: C0301: Line too long (120/100) (line-too-long),,,,
************* Module tests.unit.final_dq_checks_unit,,,,
assignment/tests/unit/final_dq_checks_unit.py:22:0: R0913: Too many arguments (12/5) (too-many-arguments),,,,
assignment/tests/unit/final_dq_checks_unit.py:22:0: R0914: Too many local variables (16/15) (too-many-locals),,,,
"assignment/tests/unit/final_dq_checks_unit.py:64:4: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/tests/unit/final_dq_checks_unit.py:130:60: E1120: No value for argument 'mock_dynamo_read_table' in function call (no-value-for-parameter),,,,
assignment/tests/unit/final_dq_checks_unit.py:130:60: E1120: No value for argument 'mock_is_s3_file' in function call (no-value-for-parameter),,,,
assignment/tests/unit/final_dq_checks_unit.py:130:60: E1120: No value for argument 'mock_read_s3_to_local' in function call (no-value-for-parameter),,,,
assignment/tests/unit/final_dq_checks_unit.py:130:60: E1120: No value for argument 'mock_write_yaml_to_s3' in function call (no-value-for-parameter),,,,
assignment/tests/unit/final_dq_checks_unit.py:170:60: E1120: No value for argument 'mock_dynamo_read_table' in function call (no-value-for-parameter),,,,
assignment/tests/unit/final_dq_checks_unit.py:170:60: E1120: No value for argument 'mock_is_s3_file' in function call (no-value-for-parameter),,,,
assignment/tests/unit/final_dq_checks_unit.py:170:60: E1120: No value for argument 'mock_read_s3_to_local' in function call (no-value-for-parameter),,,,
assignment/tests/unit/final_dq_checks_unit.py:170:60: E1120: No value for argument 'mock_write_yaml_to_s3' in function call (no-value-for-parameter),,,,
assignment/tests/unit/final_dq_checks_unit.py:208:60: E1120: No value for argument 'mock_dynamo_read_table' in function call (no-value-for-parameter),,,,
assignment/tests/unit/final_dq_checks_unit.py:208:60: E1120: No value for argument 'mock_is_s3_file' in function call (no-value-for-parameter),,,,
assignment/tests/unit/final_dq_checks_unit.py:208:60: E1120: No value for argument 'mock_read_s3_to_local' in function call (no-value-for-parameter),,,,
assignment/tests/unit/final_dq_checks_unit.py:208:60: E1120: No value for argument 'mock_write_yaml_to_s3' in function call (no-value-for-parameter),,,,
assignment/tests/unit/final_dq_checks_unit.py:276:60: E1120: No value for argument 'mock_dynamo_read_table' in function call (no-value-for-parameter),,,,
assignment/tests/unit/final_dq_checks_unit.py:276:60: E1120: No value for argument 'mock_is_s3_file' in function call (no-value-for-parameter),,,,
assignment/tests/unit/final_dq_checks_unit.py:276:60: E1120: No value for argument 'mock_read_s3_to_local' in function call (no-value-for-parameter),,,,
assignment/tests/unit/final_dq_checks_unit.py:276:60: E1120: No value for argument 'mock_write_yaml_to_s3' in function call (no-value-for-parameter),,,,
assignment/tests/unit/final_dq_checks_unit.py:329:60: E1120: No value for argument 'mock_dynamo_read_table' in function call (no-value-for-parameter),,,,
assignment/tests/unit/final_dq_checks_unit.py:329:60: E1120: No value for argument 'mock_is_s3_file' in function call (no-value-for-parameter),,,,
assignment/tests/unit/final_dq_checks_unit.py:329:60: E1120: No value for argument 'mock_read_s3_to_local' in function call (no-value-for-parameter),,,,
assignment/tests/unit/final_dq_checks_unit.py:329:60: E1120: No value for argument 'mock_write_yaml_to_s3' in function call (no-value-for-parameter),,,,
assignment/tests/unit/final_dq_checks_unit.py:366:60: E1120: No value for argument 'mock_dynamo_read_table' in function call (no-value-for-parameter),,,,
assignment/tests/unit/final_dq_checks_unit.py:366:60: E1120: No value for argument 'mock_is_s3_file' in function call (no-value-for-parameter),,,,
assignment/tests/unit/final_dq_checks_unit.py:366:60: E1120: No value for argument 'mock_read_s3_to_local' in function call (no-value-for-parameter),,,,
assignment/tests/unit/final_dq_checks_unit.py:366:60: E1120: No value for argument 'mock_write_yaml_to_s3' in function call (no-value-for-parameter),,,,
assignment/tests/unit/final_dq_checks_unit.py:438:60: E1120: No value for argument 'mock_dynamo_read_table' in function call (no-value-for-parameter),,,,
assignment/tests/unit/final_dq_checks_unit.py:438:60: E1120: No value for argument 'mock_is_s3_file' in function call (no-value-for-parameter),,,,
assignment/tests/unit/final_dq_checks_unit.py:438:60: E1120: No value for argument 'mock_read_s3_to_local' in function call (no-value-for-parameter),,,,
assignment/tests/unit/final_dq_checks_unit.py:438:60: E1120: No value for argument 'mock_write_yaml_to_s3' in function call (no-value-for-parameter),,,,
assignment/tests/unit/final_dq_checks_unit.py:477:60: E1120: No value for argument 'mock_dynamo_read_table' in function call (no-value-for-parameter),,,,
assignment/tests/unit/final_dq_checks_unit.py:477:60: E1120: No value for argument 'mock_is_s3_file' in function call (no-value-for-parameter),,,,
assignment/tests/unit/final_dq_checks_unit.py:477:60: E1120: No value for argument 'mock_read_s3_to_local' in function call (no-value-for-parameter),,,,
assignment/tests/unit/final_dq_checks_unit.py:477:60: E1120: No value for argument 'mock_write_yaml_to_s3' in function call (no-value-for-parameter),,,,
assignment/tests/unit/final_dq_checks_unit.py:514:60: E1120: No value for argument 'mock_dynamo_read_table' in function call (no-value-for-parameter),,,,
assignment/tests/unit/final_dq_checks_unit.py:514:60: E1120: No value for argument 'mock_is_s3_file' in function call (no-value-for-parameter),,,,
assignment/tests/unit/final_dq_checks_unit.py:514:60: E1120: No value for argument 'mock_read_s3_to_local' in function call (no-value-for-parameter),,,,
assignment/tests/unit/final_dq_checks_unit.py:514:60: E1120: No value for argument 'mock_write_yaml_to_s3' in function call (no-value-for-parameter),,,,
assignment/tests/unit/final_dq_checks_unit.py:569:60: E1120: No value for argument 'mock_dynamo_read_table' in function call (no-value-for-parameter),,,,
assignment/tests/unit/final_dq_checks_unit.py:569:60: E1120: No value for argument 'mock_is_s3_file' in function call (no-value-for-parameter),,,,
assignment/tests/unit/final_dq_checks_unit.py:569:60: E1120: No value for argument 'mock_read_s3_to_local' in function call (no-value-for-parameter),,,,
assignment/tests/unit/final_dq_checks_unit.py:569:60: E1120: No value for argument 'mock_write_yaml_to_s3' in function call (no-value-for-parameter),,,,
assignment/tests/unit/final_dq_checks_unit.py:613:60: E1120: No value for argument 'mock_dynamo_read_table' in function call (no-value-for-parameter),,,,
assignment/tests/unit/final_dq_checks_unit.py:613:60: E1120: No value for argument 'mock_is_s3_file' in function call (no-value-for-parameter),,,,
assignment/tests/unit/final_dq_checks_unit.py:613:60: E1120: No value for argument 'mock_read_s3_to_local' in function call (no-value-for-parameter),,,,
assignment/tests/unit/final_dq_checks_unit.py:613:60: E1120: No value for argument 'mock_write_yaml_to_s3' in function call (no-value-for-parameter),,,,
assignment/tests/unit/final_dq_checks_unit.py:655:60: E1120: No value for argument 'mock_dynamo_read_table' in function call (no-value-for-parameter),,,,
assignment/tests/unit/final_dq_checks_unit.py:655:60: E1120: No value for argument 'mock_is_s3_file' in function call (no-value-for-parameter),,,,
assignment/tests/unit/final_dq_checks_unit.py:655:60: E1120: No value for argument 'mock_read_s3_to_local' in function call (no-value-for-parameter),,,,
assignment/tests/unit/final_dq_checks_unit.py:655:60: E1120: No value for argument 'mock_write_yaml_to_s3' in function call (no-value-for-parameter),,,,
assignment/tests/unit/final_dq_checks_unit.py:6:0: W0611: Unused import datetime (unused-import),,,,
************* Module tests.unit.forced_cell_assignment_unit,,,,
assignment/tests/unit/forced_cell_assignment_unit.py:1:0: C0302: Too many lines in module (1579/1000) (too-many-lines),,,,
assignment/tests/unit/forced_cell_assignment_unit.py:10:0: E0401: Unable to import 'pyspark.sql.functions' (import-error),,,,
assignment/tests/unit/forced_cell_assignment_unit.py:23:4: W0221: Number of parameters was 1 in 'TestCase.setUp' and is now 1 in overridden 'TestForcedCellAssignment.setUp' method (arguments-differ),,,,
assignment/tests/unit/forced_cell_assignment_unit.py:62:13: W1514: Using open without explicitly specifying an encoding (unspecified-encoding),,,,
"assignment/tests/unit/forced_cell_assignment_unit.py:62:32: C0103: Variable name ""f"" doesn't conform to snake_case naming style (invalid-name)",,,,
************* Module tests.unit.main_cell_eligible_unit,,,,
assignment/tests/unit/main_cell_eligible_unit.py:7:0: E0401: Unable to import 'pyspark.sql.functions' (import-error),,,,
assignment/tests/unit/main_cell_eligible_unit.py:9:0: E0401: Unable to import 'pyspark.sql' (import-error),,,,
assignment/tests/unit/main_cell_eligible_unit.py:10:0: E0401: Unable to import 'pyspark.sql.types' (import-error),,,,
assignment/tests/unit/main_cell_eligible_unit.py:29:4: W0221: Number of parameters was 1 in 'TestCase.setUp' and is now 1 in overridden 'TestMainCellEligible.setUp' method (arguments-differ),,,,
************* Module tests.unit.main_cell_randomization_unit,,,,
assignment/tests/unit/main_cell_randomization_unit.py:162:0: C0301: Line too long (114/100) (line-too-long),,,,
assignment/tests/unit/main_cell_randomization_unit.py:163:0: C0301: Line too long (123/100) (line-too-long),,,,
assignment/tests/unit/main_cell_randomization_unit.py:534:0: C0301: Line too long (102/100) (line-too-long),,,,
assignment/tests/unit/main_cell_randomization_unit.py:628:0: C0301: Line too long (137/100) (line-too-long),,,,
assignment/tests/unit/main_cell_randomization_unit.py:10:0: E0401: Unable to import 'pyspark.sql.functions' (import-error),,,,
assignment/tests/unit/main_cell_randomization_unit.py:11:0: E0401: Unable to import 'pyspark.sql.types' (import-error),,,,
assignment/tests/unit/main_cell_randomization_unit.py:28:4: W0221: Number of parameters was 1 in 'TestCase.setUp' and is now 1 in overridden 'TestMainCellRandomization.setUp' method (arguments-differ),,,,
assignment/tests/unit/main_cell_randomization_unit.py:173:4: R0913: Too many arguments (6/5) (too-many-arguments),,,,
"assignment/tests/unit/main_cell_randomization_unit.py:192:8: C0103: Variable name ""PopulationRow"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/tests/unit/main_cell_randomization_unit.py:210:12: C0103: Variable name ""rowDict"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/tests/unit/main_cell_randomization_unit.py:175:8: W0613: Unused argument 'avg_resp_adj_score' (unused-argument),,,,
assignment/tests/unit/main_cell_randomization_unit.py:176:8: W0613: Unused argument 'avg_resp_adj_ltv_1yr_score' (unused-argument),,,,
************* Module tests.unit.model_scores_unit,,,,
assignment/tests/unit/model_scores_unit.py:8:0: E0401: Unable to import 'pyspark.sql.functions' (import-error),,,,
assignment/tests/unit/model_scores_unit.py:9:0: E0401: Unable to import 'pyspark.sql.types' (import-error),,,,
************* Module tests.unit.offer_utils_unit,,,,
assignment/tests/unit/offer_utils_unit.py:180:0: C0301: Line too long (104/100) (line-too-long),,,,
assignment/tests/unit/offer_utils_unit.py:7:0: E0401: Unable to import 'pyspark.sql.functions' (import-error),,,,
assignment/tests/unit/offer_utils_unit.py:8:0: E0401: Unable to import 'pyspark.sql.types' (import-error),,,,
assignment/tests/unit/offer_utils_unit.py:11:0: R0402: Use 'from marty.assignment.lib import pre_selection_utils' instead (consider-using-from-import),,,,
assignment/tests/unit/offer_utils_unit.py:44:20: W0622: Redefining built-in 'input' (redefined-builtin),,,,
"assignment/tests/unit/offer_utils_unit.py:80:4: C0103: Method name ""testAddedColumns"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/tests/unit/offer_utils_unit.py:97:4: C0103: Method name ""testAllocationOfOffersDistinctRoutes"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/tests/unit/offer_utils_unit.py:120:4: C0103: Method name ""testFilteredPopulation"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/tests/unit/offer_utils_unit.py:130:4: C0103: Method name ""testAllocationCounts"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/tests/unit/offer_utils_unit.py:141:12: R1716: Simplify chained comparison between the operands (chained-comparison),,,,
"assignment/tests/unit/offer_utils_unit.py:148:4: C0103: Method name ""testOfferArrayIndividual"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/tests/unit/offer_utils_unit.py:172:4: C0103: Method name ""testOfferArrayFromConfig"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/tests/unit/offer_utils_unit.py:187:4: C0103: Method name ""testForceInReasons"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/tests/unit/offer_utils_unit.py:192:8: C0103: Variable name ""a"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/tests/unit/offer_utils_unit.py:195:8: C0103: Variable name ""b"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/tests/unit/offer_utils_unit.py:219:20: W0622: Redefining built-in 'input' (redefined-builtin),,,,
assignment/tests/unit/offer_utils_unit.py:221:25: W1309: Using an f-string that does not have any interpolated variables (f-string-without-interpolation),,,,
assignment/tests/unit/offer_utils_unit.py:269:20: R1719: The if expression can be replaced with 'test' (simplifiable-if-expression),,,,
"assignment/tests/unit/offer_utils_unit.py:307:4: C0103: Method name ""testAllocationCounts"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/tests/unit/offer_utils_unit.py:318:4: C0103: Method name ""testAddedColumns"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/tests/unit/offer_utils_unit.py:336:4: C0103: Method name ""testForceInReasons"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/tests/unit/offer_utils_unit.py:341:8: C0103: Variable name ""a"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/tests/unit/offer_utils_unit.py:344:8: C0103: Variable name ""b"" doesn't conform to snake_case naming style (invalid-name)",,,,
************* Module tests.unit.pcoa_utils_unit,,,,
assignment/tests/unit/pcoa_utils_unit.py:8:0: E0401: Unable to import 'pyspark.sql.functions' (import-error),,,,
assignment/tests/unit/pcoa_utils_unit.py:9:0: E0401: Unable to import 'pyspark.sql.types' (import-error),,,,
assignment/tests/unit/pcoa_utils_unit.py:27:4: W0221: Number of parameters was 1 in 'TestCase.setUp' and is now 1 in overridden 'TestPcoaUtils.setUp' method (arguments-differ),,,,
"assignment/tests/unit/pcoa_utils_unit.py:101:4: E0213: Method should have ""self"" as first argument (no-self-argument)",,,,
"assignment/tests/unit/pcoa_utils_unit.py:118:4: E0213: Method should have ""self"" as first argument (no-self-argument)",,,,
"assignment/tests/unit/pcoa_utils_unit.py:137:4: E0213: Method should have ""self"" as first argument (no-self-argument)",,,,
"assignment/tests/unit/pcoa_utils_unit.py:154:4: E0213: Method should have ""self"" as first argument (no-self-argument)",,,,
************* Module tests.unit.pre_selection_intermediate_unit,,,,
assignment/tests/unit/pre_selection_intermediate_unit.py:1:0: C0302: Too many lines in module (1173/1000) (too-many-lines),,,,
assignment/tests/unit/pre_selection_intermediate_unit.py:428:23: W1401: Anomalous backslash in string: '\-'. String constant might be missing an r prefix. (anomalous-backslash-in-string),,,,
assignment/tests/unit/pre_selection_intermediate_unit.py:428:27: W1401: Anomalous backslash in string: '\/'. String constant might be missing an r prefix. (anomalous-backslash-in-string),,,,
assignment/tests/unit/pre_selection_intermediate_unit.py:445:23: W1401: Anomalous backslash in string: '\-'. String constant might be missing an r prefix. (anomalous-backslash-in-string),,,,
assignment/tests/unit/pre_selection_intermediate_unit.py:445:27: W1401: Anomalous backslash in string: '\/'. String constant might be missing an r prefix. (anomalous-backslash-in-string),,,,
assignment/tests/unit/pre_selection_intermediate_unit.py:8:0: E0401: Unable to import 'pyspark.sql.functions' (import-error),,,,
assignment/tests/unit/pre_selection_intermediate_unit.py:9:0: E0401: Unable to import 'pyspark.sql.types' (import-error),,,,
"assignment/tests/unit/pre_selection_intermediate_unit.py:69:4: C0103: Method name ""testForceInFreq"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/tests/unit/pre_selection_intermediate_unit.py:92:8: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/tests/unit/pre_selection_intermediate_unit.py:116:12: R1729: Use a generator instead 'all(a == b for (a, b) in zip(results, expected))' (use-a-generator),,
assignment/tests/unit/pre_selection_intermediate_unit.py:186:4: R0201: Method could be a function (no-self-use),,,,
"assignment/tests/unit/pre_selection_intermediate_unit.py:278:8: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/tests/unit/pre_selection_intermediate_unit.py:875:4: R0201: Method could be a function (no-self-use),,,,
************* Module tests.unit.pre_selection_unit,,,,
assignment/tests/unit/pre_selection_unit.py:47:0: C0301: Line too long (116/100) (line-too-long),,,,
assignment/tests/unit/pre_selection_unit.py:48:0: C0301: Line too long (116/100) (line-too-long),,,,
assignment/tests/unit/pre_selection_unit.py:75:0: C0301: Line too long (137/100) (line-too-long),,,,
assignment/tests/unit/pre_selection_unit.py:78:0: C0301: Line too long (130/100) (line-too-long),,,,
assignment/tests/unit/pre_selection_unit.py:81:0: C0301: Line too long (123/100) (line-too-long),,,,
assignment/tests/unit/pre_selection_unit.py:262:0: C0301: Line too long (109/100) (line-too-long),,,,
assignment/tests/unit/pre_selection_unit.py:265:0: C0301: Line too long (117/100) (line-too-long),,,,
assignment/tests/unit/pre_selection_unit.py:286:0: C0301: Line too long (101/100) (line-too-long),,,,
assignment/tests/unit/pre_selection_unit.py:8:0: R0402: Use 'from unittest import mock' instead (consider-using-from-import),,,,
assignment/tests/unit/pre_selection_unit.py:13:0: E0401: Unable to import 'pyspark.sql.functions' (import-error),,,,
assignment/tests/unit/pre_selection_unit.py:14:0: E0401: Unable to import 'pyspark.sql.types' (import-error),,,,
assignment/tests/unit/pre_selection_unit.py:17:0: R0402: Use 'from marty.assignment.scripts import pre_selection' instead (consider-using-from-import),,,,
assignment/tests/unit/pre_selection_unit.py:18:0: R0402: Use 'from marty.lib import test_utils' instead (consider-using-from-import),,,,
assignment/tests/unit/pre_selection_unit.py:65:21: W0109: Duplicate key 'preselected_population_recent' in dictionary (duplicate-key),,,,
"assignment/tests/unit/pre_selection_unit.py:131:20: C0103: Variable name ""c"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/tests/unit/pre_selection_unit.py:154:56: C0103: Argument name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/tests/unit/pre_selection_unit.py:235:13: W0612: Unused variable 'mock_read' (unused-variable),,,,
assignment/tests/unit/pre_selection_unit.py:237:13: W0612: Unused variable 'mock_write' (unused-variable),,,,
assignment/tests/unit/pre_selection_unit.py:239:13: W0612: Unused variable 'mock_config' (unused-variable),,,,
assignment/tests/unit/pre_selection_unit.py:243:13: W0612: Unused variable 'mock_date_info' (unused-variable),,,,
assignment/tests/unit/pre_selection_unit.py:245:13: W0612: Unused variable 'mock_get_slim' (unused-variable),,,,
assignment/tests/unit/pre_selection_unit.py:247:13: W0612: Unused variable 'mock_generate' (unused-variable),,,,
assignment/tests/unit/pre_selection_unit.py:355:12: R1716: Simplify chained comparison between the operands (chained-comparison),,,,
assignment/tests/unit/pre_selection_unit.py:526:15: W0212: Access to a protected member _get_slim_output of a client class (protected-access),,,,
assignment/tests/unit/pre_selection_unit.py:544:15: W0212: Access to a protected member _get_slim_output of a client class (protected-access),,,,
************* Module tests.unit.second_suppression_and_segmentation_unit,,,,
assignment/tests/unit/second_suppression_and_segmentation_unit.py:9:0: E0401: Unable to import 'pyspark.sql.functions' (import-error),,,,
assignment/tests/unit/second_suppression_and_segmentation_unit.py:10:0: E0401: Unable to import 'pyspark.sql.types' (import-error),,,,
assignment/tests/unit/second_suppression_and_segmentation_unit.py:34:4: W0221: Number of parameters was 1 in 'TestCase.setUp' and is now 1 in overridden 'TestFinalPreselectionUtils.setUp' method (arguments-differ),,,,
"assignment/tests/unit/second_suppression_and_segmentation_unit.py:302:4: E0213: Method should have ""self"" as first argument (no-self-argument)",,,,
"assignment/tests/unit/second_suppression_and_segmentation_unit.py:362:4: E0213: Method should have ""self"" as first argument (no-self-argument)",,,,
"assignment/tests/unit/second_suppression_and_segmentation_unit.py:395:8: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/tests/unit/second_suppression_and_segmentation_unit.py:426:4: E0213: Method should have ""self"" as first argument (no-self-argument)",,,,
"assignment/tests/unit/second_suppression_and_segmentation_unit.py:434:8: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/tests/unit/second_suppression_and_segmentation_unit.py:459:4: E0213: Method should have ""self"" as first argument (no-self-argument)",,,,
"assignment/tests/unit/second_suppression_and_segmentation_unit.py:466:8: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/tests/unit/second_suppression_and_segmentation_unit.py:514:4: E0213: Method should have ""self"" as first argument (no-self-argument)",,,,
"assignment/tests/unit/second_suppression_and_segmentation_unit.py:520:8: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/tests/unit/second_suppression_and_segmentation_unit.py:550:4: E0213: Method should have ""self"" as first argument (no-self-argument)",,,,
"assignment/tests/unit/second_suppression_and_segmentation_unit.py:556:8: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/tests/unit/second_suppression_and_segmentation_unit.py:605:4: E0213: Method should have ""self"" as first argument (no-self-argument)",,,,
"assignment/tests/unit/second_suppression_and_segmentation_unit.py:609:8: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/tests/unit/second_suppression_and_segmentation_unit.py:634:4: E0213: Method should have ""self"" as first argument (no-self-argument)",,,,
"assignment/tests/unit/second_suppression_and_segmentation_unit.py:731:8: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/tests/unit/second_suppression_and_segmentation_unit.py:748:4: W0221: Number of parameters was 1 in 'TestCase.setUp' and is now 1 in overridden 'TestRedoingForceInUtuc.setUp' method (arguments-differ),,,,
"assignment/tests/unit/second_suppression_and_segmentation_unit.py:795:4: C0103: Method name ""testUtucAfterSecondSuppression"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/tests/unit/second_suppression_and_segmentation_unit.py:810:4: C0103: Method name ""testForceInOutOnSecondSuppression"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/tests/unit/second_suppression_and_segmentation_unit.py:874:8: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/tests/unit/second_suppression_and_segmentation_unit.py:7:0: W0611: Unused patch imported from unittest.mock (unused-import),,,,
************* Module tests.unit.segments_unit,,,,
assignment/tests/unit/segments_unit.py:1043:0: C0301: Line too long (124/100) (line-too-long),,,,
assignment/tests/unit/segments_unit.py:1216:0: C0301: Line too long (106/100) (line-too-long),,,,
assignment/tests/unit/segments_unit.py:1:0: C0302: Too many lines in module (1518/1000) (too-many-lines),,,,
assignment/tests/unit/segments_unit.py:10:0: E0401: Unable to import 'pyspark.sql.functions' (import-error),,,,
assignment/tests/unit/segments_unit.py:12:0: E0401: Unable to import 'pyspark.sql.utils' (import-error),,,,
assignment/tests/unit/segments_unit.py:25:4: W0221: Number of parameters was 1 in 'TestCase.setUp' and is now 1 in overridden 'TestSegments.setUp' method (arguments-differ),,,,
"assignment/tests/unit/segments_unit.py:72:12: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/tests/unit/segments_unit.py:72:12: W0612: Unused variable 'df' (unused-variable),,,,
assignment/tests/unit/segments_unit.py:19:0: R0904: Too many public methods (36/20) (too-many-public-methods),,,,
************* Module tests.unit.uplift_utils_unit,,,,
assignment/tests/unit/uplift_utils_unit.py:8:0: E0401: Unable to import 'pyspark.sql.types' (import-error),,,,
assignment/tests/unit/uplift_utils_unit.py:21:4: W0221: Number of parameters was 1 in 'TestCase.setUp' and is now 1 in overridden 'TestUpliftUtils.setUp' method (arguments-differ),,,,
"assignment/tests/unit/uplift_utils_unit.py:135:4: E0213: Method should have ""self"" as first argument (no-self-argument)",,,,
"assignment/tests/unit/uplift_utils_unit.py:141:8: C0103: Variable name ""df"" doesn't conform to snake_case naming style (invalid-name)",,,,
************* Module tests.unit.utuc_unit,,,,
assignment/tests/unit/utuc_unit.py:11:0: E0401: Unable to import 'pyspark.sql.functions' (import-error),,,,
assignment/tests/unit/utuc_unit.py:12:0: E0401: Unable to import 'pyspark.sql.types' (import-error),,,,
assignment/tests/unit/utuc_unit.py:16:0: R0402: Use 'from marty.lib import test_utils' instead (consider-using-from-import),,,,
"assignment/tests/unit/utuc_unit.py:20:0: C0103: Class name ""generateUTUCpop"" doesn't conform to PascalCase naming style (invalid-name)",,,,
assignment/tests/unit/utuc_unit.py:42:4: R0201: Method could be a function (no-self-use),,,,
assignment/tests/unit/utuc_unit.py:70:4: R0201: Method could be a function (no-self-use),,,,
assignment/tests/unit/utuc_unit.py:99:24: R1721: Unnecessary use of a comprehension, use list(range(1, 1 + total_numbers)) instead. (unnecessary-comprehension),,
assignment/tests/unit/utuc_unit.py:87:4: R0201: Method could be a function (no-self-use),,,,
assignment/tests/unit/utuc_unit.py:138:25: E1121: Too many positional arguments for method call (too-many-function-args),,,,
assignment/tests/unit/utuc_unit.py:141:21: E1121: Too many positional arguments for method call (too-many-function-args),,,,
assignment/tests/unit/utuc_unit.py:176:25: E1121: Too many positional arguments for method call (too-many-function-args),,,,
assignment/tests/unit/utuc_unit.py:179:21: E1121: Too many positional arguments for method call (too-many-function-args),,,,
assignment/tests/unit/utuc_unit.py:211:25: E1121: Too many positional arguments for method call (too-many-function-args),,,,
assignment/tests/unit/utuc_unit.py:214:21: E1121: Too many positional arguments for method call (too-many-function-args),,,,
assignment/tests/unit/utuc_unit.py:231:4: C0202: Class method generate_utuc_input_pop should have 'cls' as first argument (bad-classmethod-argument),,,,
"assignment/tests/unit/utuc_unit.py:422:12: C0103: Variable name ""x"" doesn't conform to snake_case naming style (invalid-name)",,,,
"assignment/tests/unit/utuc_unit.py:442:12: C0103: Variable name ""x"" doesn't conform to snake_case naming style (invalid-name)",,,,
assignment/tests/unit/utuc_unit.py:521:16: W0613: Unused argument 'job' (unused-argument),,,,
assignment/tests/unit/utuc_unit.py:665:16: W0613: Unused argument 'job' (unused-argument),,,,
assignment/tests/unit/utuc_unit.py:495:12: W0201: Attribute 'output_2' defined outside __init__ (attribute-defined-outside-init),,,,
assignment/tests/unit/utuc_unit.py:525:12: W0201: Attribute 'output_zero' defined outside __init__ (attribute-defined-outside-init),,,,
assignment/tests/unit/utuc_unit.py:556:12: W0201: Attribute 'output_zero' defined outside __init__ (attribute-defined-outside-init),,,,
assignment/tests/unit/utuc_unit.py:669:12: W0201: Attribute 'output_zero' defined outside __init__ (attribute-defined-outside-init),,,,
assignment/tests/unit/utuc_unit.py:1:0: R0801: Similar lines in 2 files,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
,,,,
